{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "# Exploration of the modulators and downstream effectors of HDAC6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "import time\n",
    "import sys\n",
    "import getpass\n",
    "from collections import defaultdict\n",
    "\n",
    "import bel_repository\n",
    "import bio2bel_hgnc\n",
    "import bio2bel_famplex\n",
    "import pandas as pd\n",
    "import pybel\n",
    "import pybel_jupyter\n",
    "import pybel_tools\n",
    "import hbp_knowledge\n",
    "from pybel.dsl import Protein\n",
    "from pybel.struct import get_subgraph_by_neighborhood, get_subgraph_by_second_neighbors\n",
    "from pybel.struct.mutation import (\n",
    "    remove_biological_processes, remove_pathologies, remove_isolated_nodes, remove_associations,\n",
    ")\n",
    "from pybel.manager.citation_utils import get_pubmed_citation_response\n",
    "from pybel_tools.mutation import enrich_complexes, enrich_variants, expand_internal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BEL Repository: v0.0.8-dev\n",
      "PyBEL: v0.13.3-dev\n",
      "PyBEL-Jupyer: v0.2.2-dev\n",
      "PyBEL-Tools: v0.7.4-dev\n",
      "pharamcome/knowledge: v0.0.5\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(f'''BEL Repository: v{bel_repository.get_version()}\n",
    "PyBEL: v{pybel.get_version()}\n",
    "PyBEL-Jupyer: v{pybel_jupyter.get_version()}\n",
    "PyBEL-Tools: v{pybel_tools.get_version()}\n",
    "pharamcome/knowledge: v{hbp_knowledge.VERSION}\n",
    "''')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3.7.3 (default, Mar 27 2019, 09:23:39) \n",
      "[Clang 10.0.0 (clang-1000.11.45.5)]\n"
     ]
    }
   ],
   "source": [
    "print(sys.version)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thu Jul 25 13:35:19 2019\n"
     ]
    }
   ],
   "source": [
    "print(time.asctime())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cthoyt\n"
     ]
    }
   ],
   "source": [
    "print(getpass.getuser())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Human Brain Pharmacome Knowledge v0.0.5\n",
      "Number of Nodes: 5258\n",
      "Number of Edges: 15094\n",
      "Number of Citations: 340\n",
      "Number of Authors: 1910\n",
      "Network Density: 5.46E-04\n",
      "Number of Components: 37\n",
      "Number of Warnings: 35\n"
     ]
    }
   ],
   "source": [
    "graph = hbp_knowledge.repository.get_graph()\n",
    "remove_associations(graph)\n",
    "remove_pathologies(graph)\n",
    "remove_isolated_nodes(graph)\n",
    "graph.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<bio2bel_famplex.manager.Manager at 0x124143b00>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# TODO add expansion\n",
    "famplex_manager = bio2bel_famplex.Manager()\n",
    "famplex_manager"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Human Brain Pharmacome Knowledge v0.0.5\n",
      "Number of Nodes: 5258\n",
      "Number of Edges: 15094\n",
      "Number of Citations: 340\n",
      "Number of Authors: 1910\n",
      "Network Density: 5.46E-04\n",
      "Number of Components: 37\n",
      "Number of Warnings: 35\n"
     ]
    }
   ],
   "source": [
    "hgnc_manager = bio2bel_hgnc.Manager()\n",
    "hgnc_manager.normalize_genes(graph)\n",
    "graph.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<BEL p(FPLX:HDAC)>, <BEL p(hgnc:HDAC6)>, <BEL p(hgnc:HDAC3)>]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Find all HDAC* in the BEL graph\n",
    "\n",
    "hdacs = []\n",
    "for node in graph:\n",
    "    name = node.get('name')\n",
    "    if name is not None and name.startswith('HDAC'):\n",
    "        hdacs.append(node)\n",
    "        \n",
    "hdacs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None vNone\n",
      "Number of Nodes: 53\n",
      "Number of Edges: 56\n",
      "Number of Citations: 19\n",
      "Number of Authors: 123\n",
      "Network Density: 2.03E-02\n",
      "Number of Components: 3\n",
      "Number of Warnings: 0\n"
     ]
    }
   ],
   "source": [
    "#subgraph = get_subgraph_by_second_neighbors(graph, [hdac6])\n",
    "subgraph = get_subgraph_by_neighborhood(graph, hdacs)\n",
    "enrich_complexes(subgraph)\n",
    "enrich_variants(subgraph)\n",
    "#expand_internal(graph, subgraph)\n",
    "subgraph.summarize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/javascript": [
       "function init_d3_force(d3, graph, chart, width, height, function_colors) {\n",
       "    var focus_node = null;\n",
       "    var highlight_node = null;\n",
       "\n",
       "    // Highlight color variables\n",
       "\n",
       "    // Highlight color of the node boundering\n",
       "    const highlight_node_boundering = \"#4EB2D4\";\n",
       "\n",
       "    // Highlight color of the edge\n",
       "    const highlighted_link_color = \"#4EB2D4\";\n",
       "\n",
       "    // Text highlight color\n",
       "    const highlight_text = \"#4EB2D4\";\n",
       "\n",
       "    // Size when zooming scale\n",
       "    var size = d3.scalePow().exponent(1)\n",
       "        .domain([1, 100])\n",
       "        .range([8, 24]);\n",
       "\n",
       "    // Simulation parameters\n",
       "    const linkDistance = 100;\n",
       "    const fCharge = -1000;\n",
       "    const linkStrength = 0.7;\n",
       "    const collideStrength = 1;\n",
       "\n",
       "    // Simulation defined with variables\n",
       "    var simulation = d3.forceSimulation()\n",
       "        .force(\"link\", d3.forceLink()\n",
       "            .distance(linkDistance)\n",
       "            .strength(linkStrength)\n",
       "        )\n",
       "        .force(\"collide\", d3.forceCollide()\n",
       "            .radius(function (d) {\n",
       "                return d.r + 10\n",
       "            })\n",
       "            .strength(collideStrength)\n",
       "        )\n",
       "        .force(\"charge\", d3.forceManyBody()\n",
       "            .strength(fCharge)\n",
       "        )\n",
       "        .force(\"center\", d3.forceCenter(width / 2, height / 2))\n",
       "        .force(\"y\", d3.forceY(0))\n",
       "        .force(\"x\", d3.forceX(0));\n",
       "\n",
       "    // Pin down functionality\n",
       "    var node_drag = d3.drag()\n",
       "        .on(\"start\", dragstarted)\n",
       "        .on(\"drag\", dragged)\n",
       "        .on(\"end\", dragended);\n",
       "\n",
       "    function dragstarted(d) {\n",
       "        if (!d3.event.active) simulation.alphaTarget(0.3).restart();\n",
       "        d.fx = d.x;\n",
       "        d.fy = d.y;\n",
       "    }\n",
       "\n",
       "    function dragged(d) {\n",
       "        d.fx = d3.event.x;\n",
       "        d.fy = d3.event.y;\n",
       "    }\n",
       "\n",
       "    function dragended(d) {\n",
       "        if (!d3.event.active) simulation.alphaTarget(0);\n",
       "    }\n",
       "\n",
       "    function releasenode(d) {\n",
       "        d.fx = null;\n",
       "        d.fy = null;\n",
       "    }\n",
       "\n",
       "    //END Pin down functionality\n",
       "\n",
       "\n",
       "    /**\n",
       "     * Gets the best name for a node object\n",
       "     * @param {object} d object\n",
       "     * @returns {str} canonical name of the node\n",
       "     */\n",
       "    function getCanonicalName(d) {\n",
       "        if (d.name && !(d.variants || d.reactants || d.products || d.members)) {\n",
       "            return d.name\n",
       "        } else if (d.bel) {\n",
       "            return d.bel\n",
       "        } else {\n",
       "            console.log('Undefined node: ' + d);\n",
       "            return 'UNDEFINED'\n",
       "        }\n",
       "    }\n",
       "\n",
       "    const color_circunferencia = \"black\";\n",
       "    const default_link_color = \"#AAAAAA\";\n",
       "    const nominal_base_node_size = 8;\n",
       "\n",
       "    // Normal and highlighted stroke of the links (double the width of the link when highlighted)\n",
       "    const nominal_stroke = 1.5;\n",
       "\n",
       "    // Zoom variables\n",
       "    const min_zoom = 0.1;\n",
       "    const max_zoom = 10;\n",
       "\n",
       "    var svg = d3.select(chart).append(\"svg\")\n",
       "        .attr(\"width\", width)\n",
       "        .attr(\"height\", height);\n",
       "\n",
       "    // // Create definition for arrowhead.\n",
       "    svg.append(\"defs\").append(\"marker\")\n",
       "        .attr(\"id\", \"arrowhead\")\n",
       "        .attr(\"viewBox\", \"0 -5 10 10\")\n",
       "        .attr(\"refX\", 20)\n",
       "        .attr(\"refY\", 0)\n",
       "        .attr(\"markerUnits\", \"strokeWidth\")\n",
       "        .attr(\"markerWidth\", 6)\n",
       "        .attr(\"markerHeight\", 6)\n",
       "        .attr(\"orient\", \"auto\")\n",
       "        .append(\"path\")\n",
       "        .attr(\"d\", \"M0,-5L10,0L0,5\");\n",
       "\n",
       "    // // Create definition for stub.\n",
       "    svg.append(\"defs\").append(\"marker\")\n",
       "        .attr(\"id\", \"stub\")\n",
       "        .attr(\"viewBox\", \"-1 -5 2 10\")\n",
       "        .attr(\"refX\", 15)\n",
       "        .attr(\"refY\", 0)\n",
       "        .attr(\"markerUnits\", \"strokeWidth\")\n",
       "        .attr(\"markerWidth\", 6)\n",
       "        .attr(\"markerHeight\", 6)\n",
       "        .attr(\"orient\", \"auto\")\n",
       "        .append(\"path\")\n",
       "        .attr(\"d\", \"M 0,0 m -1,-5 L 1,-5 L 1,5 L -1,5 Z\");\n",
       "\n",
       "    // Background\n",
       "    svg.append(\"rect\")\n",
       "        .attr(\"width\", \"100%\")\n",
       "        .attr(\"height\", \"100%\")\n",
       "        .attr(\"fill\", \"#ffffff\")\n",
       "        .style(\"pointer-events\", \"all\")\n",
       "        // Zoom + panning functionality\n",
       "        .call(d3.zoom()\n",
       "            .scaleExtent([min_zoom, max_zoom])\n",
       "            .on(\"zoom\", zoomed))\n",
       "        .on(\"dblclick.zoom\", null);\n",
       "\n",
       "\n",
       "    function zoomed() {\n",
       "        g.attr(\"transform\", d3.event.transform);\n",
       "    }\n",
       "\n",
       "    // g = svg object where the graph will be appended\n",
       "    var g = svg.append(\"g\");\n",
       "\n",
       "    var linkedByIndex = {};\n",
       "    graph.links.forEach(function (d) {\n",
       "        linkedByIndex[d.source + \",\" + d.target] = true;\n",
       "    });\n",
       "\n",
       "    function isConnected(a, b) {\n",
       "        return linkedByIndex[a.index + \",\" + b.index] || linkedByIndex[b.index + \",\" + a.index] || a.index == b.index;\n",
       "    }\n",
       "\n",
       "    function ticked() {\n",
       "        link.attr(\"x1\", function (d) {\n",
       "            return d.source.x;\n",
       "        })\n",
       "            .attr(\"y1\", function (d) {\n",
       "                return d.source.y;\n",
       "            })\n",
       "            .attr(\"x2\", function (d) {\n",
       "                return d.target.x;\n",
       "            })\n",
       "            .attr(\"y2\", function (d) {\n",
       "                return d.target.y;\n",
       "            });\n",
       "\n",
       "        node\n",
       "            .attr(\"transform\", function (d) {\n",
       "                return \"translate(\" + d.x + \", \" + d.y + \")\";\n",
       "            });\n",
       "    }\n",
       "\n",
       "\n",
       "    simulation\n",
       "        .nodes(graph.nodes)\n",
       "        .on(\"tick\", ticked);\n",
       "\n",
       "    simulation.force(\"link\")\n",
       "        .links(graph.links);\n",
       "\n",
       "    // Definition of links nodes text...\n",
       "\n",
       "    var link = g.selectAll(\".link\")\n",
       "        .data(graph.links)\n",
       "        .enter().append(\"line\")\n",
       "        .style(\"stroke-width\", nominal_stroke)\n",
       "        .style(\"stroke\", default_link_color)\n",
       "        .style(\"stroke-dasharray\", function (d) {\n",
       "            if (['decreases', 'directlyDecreases', 'increases', 'directlyIncreases', 'negativeCorrelation',\n",
       "                'positiveCorrelation'].indexOf(d.relation) >= 0) {\n",
       "                return \"none\"\n",
       "            } else {\n",
       "                return \"4, 4\"\n",
       "            }\n",
       "        })\n",
       "        .attr(\"marker-start\", function (d) {\n",
       "            if ('positiveCorrelation' == d.relation) {\n",
       "                return \"url(#arrowhead)\"\n",
       "            }\n",
       "            else if ('negativeCorrelation' == d.relation) {\n",
       "                return \"url(#stub)\"\n",
       "            }\n",
       "            else {\n",
       "                return \"\"\n",
       "            }\n",
       "        })\n",
       "        .attr(\"marker-end\", function (d) {\n",
       "            if (['increases', 'directlyIncreases', 'positiveCorrelation', 'isA', 'partOf'].indexOf(d.relation) >= 0) {\n",
       "                return \"url(#arrowhead)\"\n",
       "            } else if (['decreases', 'directlyDecreases', 'negativeCorrelation'].indexOf(d.relation) >= 0) {\n",
       "                return \"url(#stub)\"\n",
       "            } else {\n",
       "                return \"\"\n",
       "            }\n",
       "        });\n",
       "\n",
       "    var node = g.selectAll(\".nodes\")\n",
       "        .data(graph.nodes)\n",
       "        .enter().append(\"g\")\n",
       "        .attr(\"class\", \"node\")\n",
       "        // Next two lines -> Pin down functionality\n",
       "        .on('dblclick', releasenode)\n",
       "        .call(node_drag);\n",
       "\n",
       "    var circle = node.append(\"path\")\n",
       "        .attr(\"d\", d3.symbol()\n",
       "            .size(function (d) {\n",
       "                return Math.PI * Math.pow(size(d.size) || nominal_base_node_size, 2);\n",
       "            })\n",
       "        )\n",
       "        .attr(\"class\", function (d) {\n",
       "            return d.function\n",
       "        })\n",
       "        .style('fill', function (d) {\n",
       "            return function_colors[d.function]\n",
       "        })\n",
       "        .style(\"stroke-width\", nominal_stroke)\n",
       "        .style(\"stroke\", color_circunferencia);\n",
       "\n",
       "    var text = node.append(\"text\")\n",
       "        .attr(\"class\", \"node-name\")\n",
       "        // .attr(\"id\", nodehashes[d])\n",
       "        .attr(\"fill\", \"black\")\n",
       "        .attr(\"dx\", 12)\n",
       "        .attr(\"dy\", \".35em\")\n",
       "        .text(function (d) {\n",
       "            return getCanonicalName(d)\n",
       "        });\n",
       "\n",
       "    // Highlight on mouseenter and back to normal on mouseout\n",
       "    node.on(\"mouseenter\", function (d) {\n",
       "        set_highlight(d);\n",
       "    })\n",
       "        .on(\"mousedown\", function () {\n",
       "            d3.event.stopPropagation();\n",
       "        }).on(\"mouseout\", function () {\n",
       "        exit_highlight();\n",
       "    });\n",
       "\n",
       "    function exit_highlight() {\n",
       "        highlight_node = null;\n",
       "        if (focus_node === null) {\n",
       "            if (highlight_node_boundering != color_circunferencia) {\n",
       "                circle.style(\"stroke\", color_circunferencia);\n",
       "                text.style(\"fill\", \"black\");\n",
       "                link.style(\"stroke\", default_link_color);\n",
       "            }\n",
       "        }\n",
       "    }\n",
       "\n",
       "    function set_highlight(d) {\n",
       "        if (focus_node !== null) d = focus_node;\n",
       "        highlight_node = d;\n",
       "\n",
       "        if (highlight_node_boundering != color_circunferencia) {\n",
       "            circle.style(\"stroke\", function (o) {\n",
       "                return isConnected(d, o) ? highlight_node_boundering : color_circunferencia;\n",
       "            });\n",
       "            text.style(\"fill\", function (o) {\n",
       "                return isConnected(d, o) ? highlight_text : \"black\";\n",
       "            });\n",
       "            link.style(\"stroke\", function (o) {\n",
       "                return o.source.index == d.index || o.target.index == d.index ? highlighted_link_color : default_link_color;\n",
       "            });\n",
       "        }\n",
       "    }\n",
       "\n",
       "\n",
       "    // Freeze the graph when space is pressed\n",
       "    function freezeGraph() {\n",
       "        // Space button Triggers STOP\n",
       "        if (d3.event.keyCode == 32) {\n",
       "            simulation.stop();\n",
       "        }\n",
       "    }\n",
       "\n",
       "    // Call freezeGraph when a key is pressed, freezeGraph checks whether this key is \"Space\" that triggers the freeze\n",
       "    d3.select(window).on(\"keydown\", freezeGraph);\n",
       "}\n",
       "\n",
       "require.config({\n",
       "    paths: {\n",
       "        d3: '//cdnjs.cloudflare.com/ajax/libs/d3/4.5.0/d3.min'\n",
       "    }\n",
       "});\n",
       "\n",
       "var elementInnerHTML = \"<div id='uzkoxrqtjpghsaec'></div>\";\n",
       "\n",
       "element.append(elementInnerHTML);\n",
       "\n",
       "require(['d3'], function (d3) {\n",
       "    return init_d3_force(d3, {\"directed\": true, \"graph\": {\"annotation_list\": {\"AssayType\": [\"B\"], \"Braak_Stage\": [\"Stage I\", \"Stage II\", \"Stage III\", \"Stage IV\", \"Stage V\", \"Stage VI\"], \"Cell_Line\": [\"BV2\", \"CG-4\", \"CHO\", \"COS7\", \"Chicken embryo fibroblasts\", \"HCN2A\", \"HEK293\", \"HEK293T\", \"HNPC\", \"HeLa\", \"Human fibroblasts\", \"N2a\", \"NIH/3T3\", \"NT2\", \"PC12\", \"SH-SY5Y\", \"SK-N-SH\", \"neural stem cell\"], \"Disease_Progression\": [\"Early Stage\", \"Late Stage\"], \"Duration\": [\"2 hours\"], \"Enzyme_Acitvity\": [\"1 U/ml\"], \"ExpMethod\": [\"in vitro\", \"in vivo\"], \"GOCC\": [\"axon cytoplasm\", \"membrane raft\", \"neuron projection\"], \"HBP_Disease\": [\"Autism Spectrum Disorder (ASD)\", \"Corticobasal Degeneration\", \"Frontotemporal Lobar Degeneration\", \"Lewy Body Disease, Diffuse\"], \"Half_life\": [\"29 min\", \"30 min\", \"70 min\"], \"HasQuantitativeInformation\": [\"False\", \"True\"], \"IC50\": [\"0.2 mM\", \"0.4 mM\", \"0.4 mg/ml\", \"0.6 mM\", \"160 mM\", \"19 nM\", \"2 mM\", \"30 nM\", \"35 mM\", \"38 nM\", \"4.5 mM\", \"45 nM\", \"450 nM\", \"5-50 nM\", \"50 mM\", \"54 nM\", \"62 nM\", \"65 nM\", \"78 nM\", \"90 nM\"], \"Ki\": [\"110 nM\", \"210 nM\", \"38 μM\", \"39 nM\", \"4-80 nM\", \"40 nM\", \"5 nM\", \"8 μM\"], \"Km\": [\"114 μM\", \"16 μM\", \"2.7 µM\", \"200 μM\", \"5 μM\", \"59 μM\", \"8-15 μM\"], \"Km_ATP\": [\"10 μM\", \"2 μM\"], \"MeasurementRelation\": [\"=\", \"~\"], \"MeasurementType\": [\"CSF concentration\", \"IC50\", \"Ki\", \"brain concertration\", \"dose\", \"plasma concentration\"], \"MeasurementUnits\": [\"mg\", \"nM\", \"μM\"], \"MeasurementValue\": [\"1.3\", \"1.4\", \"1.9\"], \"Method\": [\"Atomic Force Microscopy\", \"Bradford Assay\", \"CD Spectroscopy\", \"Cell Viability Assay, MTT\", \"Chromatin Immunoprecipitation\", \"Chromatography\", \"Chromatography, Affinity\", \"Chromatography, FPLC\", \"Chromatography, Gel filtration\", \"Chromatography, Immunoaffinity\", \"Chromatography, Ni-NTA-Sepharose\", \"Chymotrypsin Assay\", \"Co-Immunoprecipitation\", \"Comet Assay\", \"Confocal Laser Microscopy\", \"Confocal Microscopy\", \"Confocal Microscopy, Laser Scanning\", \"Deglycosylation Assay\", \"Degradation Assay, 20S Proteasomal\", \"Dephosphorylation Assay\", \"Dephosphorylation Assay, Colorimetric\", \"Dot Blot\", \"Dot blot\", \"ELISA\", \"Electron Microscopy\", \"Electron Microscopy, Immunological\", \"Electron Microscopy, Transmission\", \"Electrophoretic Mobility Shift Assay\", \"Enhanced Chemioluminescence visualization (ECL)\", \"F-actin depolymerization assay\", \"Flow Cytometry\", \"Fluorescence Microscopy\", \"Fluorescence Microscopy, Confocal\", \"Fluorescence Resonance Energy Transfer\", \"Fluorescence Spectrophotometry\", \"Immunoaffinity Chromatography\", \"Immunoaffinity purification\", \"Immunoblotting\", \"Immunocytochemistry\", \"Immunofluorescence\", \"Immunofluorescence Imaging\", \"Immunohistochemistry\", \"Immunoprecipitation\", \"Imunofluorescence\", \"In vitro actin-bundling assay\", \"In-vivo two photon calcium imaging\", \"LDH Assay\", \"Laser Capture Microdissection\", \"Liquid Chromatography\", \"Mass Spectrometry\", \"Microarray\", \"Microtubule Co-Sedimentation Assay\", \"Microtubule Sedimentation Assay\", \"Microtubule polymerization Assay\", \"Microtubule polzmerization Assay\", \"Microtubule-binding Assay\", \"NBT colorimetric fructosamine assay\", \"NMR Spectroscopy\", \"Non-denaturing Fast Halo Assay\", \"Phosphorylation Assay\", \"Protease-coupled Assay\", \"Pull-down Assay\", \"Pull-down Assay, GFP\", \"Pull-down Assay, GST\", \"Pull-down Assay, His-tagged\", \"Pull-down assay, GST\", \"Pull-down assay, His-tagged\", \"RNA Interference, shRNA\", \"RNA Interference, siRNA\", \"RP HPLC\", \"RT-PCR\", \"SDS-PAGE\", \"Scanning electron microscopy (SEM)\", \"Surface Plasmon Resonance\", \"Tandem Affinity Purification, TAP\", \"Tau aggregation assay\", \"Terminal Continuation RNA Amplification\", \"Thioflavin T Assay\", \"Thioflavin T Fluorescence Assay\", \"Transglutaminase Assay\", \"Transmission electron microscopy (TEM)\", \"Tubulin polymerization assay, light scattering method\", \"Turbidometric Analysis\", \"Ubiquitination Assay\", \"Western Blot\", \"Yeast Two-Hybrid Assay\"], \"MouseStrain\": [\"K3 model with hyperphoshorylated tau resembling FTD\", \"Tau X Drp1+/-\", \"rTg4510\"], \"PublicationType\": [\"Comment\", \"Research\", \"Review\"], \"Research_Model\": [\"5xFAD mice\", \"APP/PS1 x rTg4510 mice\", \"APPPS1 mice\", \"APPswe x P301L transgenic mice\", \"E4FAD mice\", \"HD murine model\", \"K3 model with hyperphoshorylated tau resembling FTD\", \"LRRK2 transgenic mice\", \"P301S mice\", \"PS19 mice\", \"SAMP8 mice\", \"TPR50 mice\", \"Ts65Dn mice\", \"hTau E391 transgenic mice\", \"p25 transgenic mice\", \"pR5 mice\", \"rTg4510 mice\", \"streptozotocin (STZ)-induced DB murine model\"], \"Study_Group\": [\"Han chinese\", \"Korean population\", \"Spanish cohort\", \"Tet-mev-1 mice\"], \"Substrate_Concentration\": [\"0.5 M\"], \"TauIsoform\": [\"Isoform F (441 aa)\"], \"TauIsoformGradient\": [\"3R-High\", \"3R-Low\", \"3R-Medium\", \"3R-Very High\", \"4R-High\", \"4R-Low\", \"4R-Medium\", \"4R-Very High\"], \"Tau_Antibody\": [\"9G3\", \"Anti-pS422\", \"TN1\", \"TOC1\", \"Tau-12\", \"Tau-13\", \"Tau-5\", \"TauC3\", \"pY18\"], \"Tau_Domain\": [\"proline-rich domain\"], \"Tau_Epitope\": [\"AD2\", \"AP422\", \"AT270\", \"AT8\"], \"Tau_Motif\": [\"KXGS\", \"PHF6\"], \"Tau_State\": [\"Paired Helical Filament\"], \"Tau_Structure\": [\"microtubule-binding region\", \"projection domain\"], \"Vmax\": [\"0.8-4 μM/min/mg\", \"106 nM/min/mg\", \"140 nM/min/mg\", \"2400 nM/min/mg\", \"59 μM\", \"650 nM/min/mg\", \"656 nM/min/mg\", \"80 nM/min/mg\"]}, \"annotation_pattern\": {\"ArticleCompoundNumber\": \".*\", \"Figure\": \".*\", \"MeasurementRangeLower\": \".*\", \"MeasurementRangeUpper\": \".*\", \"MeasurementValue\": \".*\", \"Page\": \".*\", \"Species\": \"^\\\\d+$\"}, \"annotation_url\": {\"Anatomy\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/anatomy/anatomy-20170511.belanno\", \"Cell\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/cell/cell-20170511.belanno\", \"CellLine\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/cell-line/cell-line-20170511.belanno\", \"CellStructure\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/cell-structure/cell-structure-20170511.belanno\", \"Confidence\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/confidence/confidence-1.0.0.belanno\", \"Disease\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/disease/disease-20170511.belanno\", \"Gender\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/gender/gender-1.0.0.belanno\", \"MeSHAnatomy\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/mesh-anatomy/mesh-anatomy-20170511.belanno\", \"MeSHDisease\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/mesh-diseases/mesh-diseases-20170511.belanno\", \"Species\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/species-taxonomy-id/species-taxonomy-id-20170511.belanno\", \"Subgraph\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/neurommsig/neurommsig-1.0.3.belanno\", \"TextLocation\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/text-location/text-location-1.0.1.belanno\"}, \"document_metadata\": {}, \"namespace_pattern\": {\"CL\": \"^\\\\d{7}$\", \"DBSNP\": \"rs[0-9]+\", \"DNSNP\": \"rs[0-9]+\", \"ECCODE\": \"^\\\\d+\\\\.-\\\\.-\\\\.-|\\\\d+\\\\.\\\\d+\\\\.-\\\\.-|\\\\d+\\\\.\\\\d+\\\\.\\\\d+\\\\.-|\\\\d+\\\\.\\\\d+\\\\.\\\\d+\\\\.(n)?\\\\d+$\", \"HBP\": \".*\", \"NCBIGENE\": \"^\\\\d+$\", \"PUBCHEM\": \"^\\\\d+$\", \"TAXONOMY\": \"^\\\\d+$\", \"UNIPROT\": \"^([A-N,R-Z][0-9]([A-Z][A-Z, 0-9][A-Z, 0-9][0-9]){1,2})|([O,P,Q][0-9][A-Z, 0-9][A-Z, 0-9][A-Z, 0-9][0-9])(\\\\.\\\\d+)?$\"}, \"namespace_url\": {\"CHEBI\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/chebi-names.belns\", \"DOID\": \"https://raw.githubusercontent.com/pharmacome/terminology/c0c576e3d3ab9b25a3982251a4ee2fd7c49e1679/external/doid-names.belns\", \"DRUGBANK\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/drugbank-names.belns\", \"ECCODE\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/ec-code.belns\", \"FB\": \"https://raw.githubusercontent.com/pharmacome/terminology/c328ad964c08967a0417a887510b97b965a62fa5/external/fb-names.belns\", \"FPLX\": \"https://raw.githubusercontent.com/sorgerlab/famplex/e8ae9926ff95266032cb74f77973c84939bffbeb/export/famplex.belns\", \"GO\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/go-names.belns\", \"HBP\": \"https://raw.githubusercontent.com/pharmacome/terminology/3074b85b858455d8eeb76cfcdef685ced19bbe11/export/hbp-names.belns\", \"HGNC\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/hgnc-names.belns\", \"HGNCGENEFAMILY\": \"https://raw.githubusercontent.com/pharmacome/terminology/efc856fb009a39e4d284269a6801f79ed3d3cf56/external/hgnc.genefamily-names.belns\", \"HP\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/hp/hp-20171108.belns\", \"INTERPRO\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/interpro/interpro-20170731.belns\", \"MESH\": \"https://raw.githubusercontent.com/pharmacome/terminology/1b20f0637c395f8aa89c2e2e342d7b704062c242/external/mesh-names.belns\", \"MGI\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/mgi-mouse-genes/mgi-mouse-genes-20170725.belns\", \"MIRBASE\": \"https://raw.githubusercontent.com/pharmacome/terminology/c328ad964c08967a0417a887510b97b965a62fa5/external/mirbase-names.belns\", \"NCBIGENE\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/entrez.belns\", \"PFAM\": \"https://raw.githubusercontent.com/pharmacome/terminology/8ccfed235e418e4c8aa576f9a5ef0f838e794c7f/external/pfam-names.belns\", \"RGD\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/rgd-rat-genes/rgd-rat-genes-20170725.belns\"}, \"namespaces_uncached\": [], \"pybel_version\": \"0.13.3-dev\"}, \"links\": [{\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Denner P\", \"Fava E\", \"Kitanovic A\", \"Kurkowsky B\", \"Mandelkow E\", \"Möhl C\", \"Pickhardt M\", \"Tassoni M\"], \"date\": \"2019-04-01\", \"first\": \"Pickhardt M\", \"last\": \"Mandelkow E\", \"name\": \"Neurobiology of aging\", \"pages\": \"24-34\", \"reference\": \"30640040\", \"title\": \"Screening of a neuronal cell model of tau pathology for therapeutic compounds.\", \"type\": \"PubMed\", \"volume\": \"76\"}, \"evidence\": \"Treatment with Givinostat (1.2 µM) lead to a nearly 7-fold increase in the fraction of ThS + cells (from ~ 12% to ~ 84%, Fig. S4A and B, blue lines) compared to the uninduced control, and 2.5-fold compared to the induced control. \", \"key\": \"0cc991075302c23a66928499fcb4bfb66fe9c3f9171c17034c2a7b2aa49bf01e2c46ab9aa71c9b0ce8d7a7ed9a96f1115e940cee1292f477455941e3dd9e71ea\", \"line\": 106, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"decreases\", \"source\": 15, \"target\": 30}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Denner P\", \"Fava E\", \"Kitanovic A\", \"Kurkowsky B\", \"Mandelkow E\", \"Möhl C\", \"Pickhardt M\", \"Tassoni M\"], \"date\": \"2019-04-01\", \"first\": \"Pickhardt M\", \"last\": \"Mandelkow E\", \"name\": \"Neurobiology of aging\", \"pages\": \"24-34\", \"reference\": \"30640040\", \"title\": \"Screening of a neuronal cell model of tau pathology for therapeutic compounds.\", \"type\": \"PubMed\", \"volume\": \"76\"}, \"evidence\": \" Conversely, compounds related to HDAC inhibition (16 of 20) led to enhanced Tau aggregation, suggesting that HDAC activity is important for suppressing aggregation\", \"key\": \"0056c2bd2b299d96b0de8efd59711e4178d10c001232f947db17e027fe4eb5ac7f484b125056260128bc2b76869ebb2401a31cadc9dc5a6afe9b58ee9b578184\", \"line\": 97, \"relation\": \"decreases\", \"source\": 30, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 6}, {\"annotations\": {\"Confidence\": {\"Medium\": true}, \"Species\": {\"10090\": true}}, \"citation\": {\"authors\": [\"Cuadrado-Tejedor M\", \"Franco R\", \"Garcia-Barroso C\", \"Garcia-Osta A\", \"Mederos S\", \"Oyarzabal J\", \"Perea G\", \"Pérez-González M\", \"Rabal O\", \"Segura V\", \"Sánchez-Arias JA\", \"Ugarte A\"], \"date\": \"2017-01-01\", \"first\": \"Cuadrado-Tejedor M\", \"last\": \"Garcia-Osta A\", \"name\": \"Neuropsychopharmacology : official publication of the American College of Neuropsychopharmacology\", \"pages\": \"524-539\", \"reference\": \"27550730\", \"title\": \"A First-in-Class Small-Molecule that Acts as a Dual Inhibitor of HDAC and PDE5 and that Rescues Hippocampal Synaptic Impairment in Alzheimer's Disease Mice.\", \"type\": \"PubMed\", \"volume\": \"42\"}, \"evidence\": \"Here we report the discovery of a new first-in-class small-molecule (CM-414) that acts as a dual inhibitor of PDE5 and HDACs. We have used this compound as a chemical probe to validate this systems therapeutics strategy, where an increase in the activation of cAMP/cGMP-responsive element-binding protein (CREB) induced by PDE5 inhibition, combined with moderate HDAC class I inhibition, leads to efficient histone acetylation. This molecule rescued the impaired long-term potentiation evident in hippocampal slices from APP/PS1 mice. Chronic treatment of Tg2576 mice with CM-414 diminished brain Aβ and tau phosphorylation (pTau) levels, increased the inactive form of GSK3β, reverted the decrease in dendritic spine density on hippocampal neurons, and reversed their cognitive deficits, at least in part by inducing the expression of genes related to synaptic transmission.\", \"key\": \"76854f3894cf8158ba456ae0ec67958fe9737f31fe135ac19f5573330e3854dedf9465e6be3981b3056aa70aec770518e37cafb06531f77b94536796e45650f1\", \"line\": 3802, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"negativeCorrelation\", \"source\": 30, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 29}, {\"annotations\": {\"Cell_Line\": {\"HEK293\": true}, \"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Cohen TJ\", \"Guo JL\", \"Hurtado DE\", \"Kwong LK\", \"Lee VM\", \"Mills IP\", \"Trojanowski JQ\"], \"date\": \"2011-01-01\", \"first\": \"Cohen TJ\", \"last\": \"Lee VM\", \"name\": \"Nature communications\", \"pages\": \"252\", \"reference\": \"21427723\", \"title\": \"The acetylation of tau inhibits its function and promotes pathological tau aggregation.\", \"type\": \"PubMed\", \"volume\": \"2\"}, \"evidence\": \"Although low-level tau acetylation was observed in untreated HEK-T40 cells, treatment with the pan histone deacetylase (HDAC) inhibitor trichostatin A (TSA), but not the Sir2 class inhibitor nicotinamide, resulted in a dramatic increase in acetylated tau levels.\", \"key\": \"d4c09f8cc9b0aa9f1bc92695b0b60524ff38686cd538f5d86131481508acbd9c836e6f62a87d4514e3a83eaf1c6de4216a120ea78ad1fc6f6a1cde700f691623\", \"line\": 4049, \"relation\": \"decreases\", \"source\": 30, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 47}, {\"annotations\": {\"Confidence\": {\"Medium\": true}, \"Species\": {\"10090\": true}}, \"citation\": {\"authors\": [\"Cuadrado-Tejedor M\", \"Franco R\", \"Garcia-Barroso C\", \"Garcia-Osta A\", \"Mederos S\", \"Oyarzabal J\", \"Perea G\", \"Pérez-González M\", \"Rabal O\", \"Segura V\", \"Sánchez-Arias JA\", \"Ugarte A\"], \"date\": \"2017-01-01\", \"first\": \"Cuadrado-Tejedor M\", \"last\": \"Garcia-Osta A\", \"name\": \"Neuropsychopharmacology : official publication of the American College of Neuropsychopharmacology\", \"pages\": \"524-539\", \"reference\": \"27550730\", \"title\": \"A First-in-Class Small-Molecule that Acts as a Dual Inhibitor of HDAC and PDE5 and that Rescues Hippocampal Synaptic Impairment in Alzheimer's Disease Mice.\", \"type\": \"PubMed\", \"volume\": \"42\"}, \"evidence\": \"Here we report the discovery of a new first-in-class small-molecule (CM-414) that acts as a dual inhibitor of PDE5 and HDACs. We have used this compound as a chemical probe to validate this systems therapeutics strategy, where an increase in the activation of cAMP/cGMP-responsive element-binding protein (CREB) induced by PDE5 inhibition, combined with moderate HDAC class I inhibition, leads to efficient histone acetylation. This molecule rescued the impaired long-term potentiation evident in hippocampal slices from APP/PS1 mice. Chronic treatment of Tg2576 mice with CM-414 diminished brain Aβ and tau phosphorylation (pTau) levels, increased the inactive form of GSK3β, reverted the decrease in dendritic spine density on hippocampal neurons, and reversed their cognitive deficits, at least in part by inducing the expression of genes related to synaptic transmission.\", \"key\": \"3af64b06a41c69f2836ea8d1fb1efcf2cf24a97dac43fa72134779fae34ed21787a2daa0c9992d3cb531216c0d3160e3ddafffd0e33750e17f8fd03883ec8342\", \"line\": 3802, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"negativeCorrelation\", \"source\": 29, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 30}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Battistuzzi G\", \"Giannini G\"], \"date\": \"2016-12-01\", \"first\": \"Battistuzzi G\", \"last\": \"Giannini G\", \"name\": \"Current bioactive compounds\", \"pages\": \"282-288\", \"reference\": \"27917100\", \"title\": \"Synthesis of ST7612AA1, a Novel Oral HDAC Inhibitor, via Radical  ",
       "Thioacetic Acid Addition.\", \"type\": \"PubMed\", \"volume\": \"12\"}, \"evidence\": \"In conclusion, ST7612AA1, prodrug of ST7464AA1, is the first of a new generation of HDAC inhibitors, very potent, orally administered, and well tolerated. It is a thiol derivative, pan-histone deacetylase inhibitor, active against a broad panel of cancer cell lines and in vivo tumor models.\", \"key\": \"dbebd975e23a5a93eba7d95798cf1477a2ba4c6fd73dfe644fc8812140615c8bfa6d57b30bf8b7f3ef90c5ec3224317c0fc6a6090592739b81b1c312ab7a6f2a\", \"line\": 3706, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"directlyDecreases\", \"source\": 8, \"target\": 30}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Cincinelli R\", \"Cuendet M\", \"Dallavalle S\", \"De Cesare M\", \"Giannini G\", \"Musso L\", \"Nurisso A\", \"Simoes-Pires C\", \"Zuco V\", \"Zunino F\", \"Zwick V\"], \"date\": \"2016-04-13\", \"first\": \"Cincinelli R\", \"last\": \"Dallavalle S\", \"name\": \"European journal of medicinal chemistry\", \"pages\": \"99-105\", \"reference\": \"26890116\", \"title\": \"Biphenyl-4-yl-acrylohydroxamic acids: Identification of a novel indolyl-substituted HDAC inhibitor with antitumor activity.\", \"type\": \"PubMed\", \"volume\": \"112\"}, \"evidence\": \"Modification of the cap group of biphenylacrylohydroxamic acid-based HDAC inhibitors led to the identification of a new derivative (3) characterized by an indolyl-substituted 4-phenylcinnamic skeleton.\", \"key\": \"b0f39cc5f1708dc391c04837153527cc29879a0e04bea21a182ad85f3440c3bf211612aeeb0bd88d0d2dc31f5170f69b311064ffd5ad539903a1f6d30eff240b\", \"line\": 3714, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"directlyDecreases\", \"source\": 5, \"target\": 30}, {\"annotations\": {\"Confidence\": {\"Medium\": true}, \"Species\": {\"10090\": true}}, \"citation\": {\"authors\": [\"Cuadrado-Tejedor M\", \"Franco R\", \"Garcia-Barroso C\", \"Garcia-Osta A\", \"Mederos S\", \"Oyarzabal J\", \"Perea G\", \"Pérez-González M\", \"Rabal O\", \"Segura V\", \"Sánchez-Arias JA\", \"Ugarte A\"], \"date\": \"2017-01-01\", \"first\": \"Cuadrado-Tejedor M\", \"last\": \"Garcia-Osta A\", \"name\": \"Neuropsychopharmacology : official publication of the American College of Neuropsychopharmacology\", \"pages\": \"524-539\", \"reference\": \"27550730\", \"title\": \"A First-in-Class Small-Molecule that Acts as a Dual Inhibitor of HDAC and PDE5 and that Rescues Hippocampal Synaptic Impairment in Alzheimer's Disease Mice.\", \"type\": \"PubMed\", \"volume\": \"42\"}, \"evidence\": \"Here we report the discovery of a new first-in-class small-molecule (CM-414) that acts as a dual inhibitor of PDE5 and HDACs. We have used this compound as a chemical probe to validate this systems therapeutics strategy, where an increase in the activation of cAMP/cGMP-responsive element-binding protein (CREB) induced by PDE5 inhibition, combined with moderate HDAC class I inhibition, leads to efficient histone acetylation. This molecule rescued the impaired long-term potentiation evident in hippocampal slices from APP/PS1 mice. Chronic treatment of Tg2576 mice with CM-414 diminished brain Aβ and tau phosphorylation (pTau) levels, increased the inactive form of GSK3β, reverted the decrease in dendritic spine density on hippocampal neurons, and reversed their cognitive deficits, at least in part by inducing the expression of genes related to synaptic transmission.\", \"key\": \"ab531ebdb76f48669111094fac19338edb9b2fe5628cf9c5585224b5f30b898b70285efe9d58eeee95b929a4ef1abf9ffe3e1a06ec21b96e817fe39af30d54af\", \"line\": 3800, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"directlyDecreases\", \"source\": 11, \"target\": 30}, {\"annotations\": {\"Cell_Line\": {\"HEK293\": true}, \"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Cohen TJ\", \"Guo JL\", \"Hurtado DE\", \"Kwong LK\", \"Lee VM\", \"Mills IP\", \"Trojanowski JQ\"], \"date\": \"2011-01-01\", \"first\": \"Cohen TJ\", \"last\": \"Lee VM\", \"name\": \"Nature communications\", \"pages\": \"252\", \"reference\": \"21427723\", \"title\": \"The acetylation of tau inhibits its function and promotes pathological tau aggregation.\", \"type\": \"PubMed\", \"volume\": \"2\"}, \"evidence\": \"Although low-level tau acetylation was observed in untreated HEK-T40 cells, treatment with the pan histone deacetylase (HDAC) inhibitor trichostatin A (TSA), but not the Sir2 class inhibitor nicotinamide, resulted in a dramatic increase in acetylated tau levels.\", \"key\": \"e559b7c158d048233551b42fe9668c018e0ad27a1dd2be8c49cd5405034a1f94f430aa814d6eec68c8c80df74843bfc96ed06b6470b3711ce84a9ad954eb8a29\", \"line\": 4048, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"directlyDecreases\", \"source\": 1, \"target\": 30}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Nedelsky NB\", \"Taylor JP\", \"Todd PK\"], \"date\": \"2008-12-01\", \"first\": \"Nedelsky NB\", \"last\": \"Taylor JP\", \"name\": \"Biochimica et biophysica acta\", \"pages\": \"691-9\", \"reference\": \"18930136\", \"title\": \"Autophagy and the ubiquitin-proteasome system: collaborators in neuroprotection.\", \"type\": \"PubMed\", \"volume\": \"1782\"}, \"evidence\": \"Specifically, it is suggested that K63-linked polyubiquitin chains recruit p62 and HDAC6 providing a signal for autophagic degradation [92,93].\", \"key\": \"df250e53c68735b8cc3d1c15009ffada566e4f6de669c57efd3c49d8df06ea78944d6ef690bb1bf962c284527f1c8f71a0a896c1711e4c44eeff1592319ce0ee\", \"line\": 698, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"increases\", \"source\": 20, \"target\": 45}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Nedelsky NB\", \"Taylor JP\", \"Todd PK\"], \"date\": \"2008-12-01\", \"first\": \"Nedelsky NB\", \"last\": \"Taylor JP\", \"name\": \"Biochimica et biophysica acta\", \"pages\": \"691-9\", \"reference\": \"18930136\", \"title\": \"Autophagy and the ubiquitin-proteasome system: collaborators in neuroprotection.\", \"type\": \"PubMed\", \"volume\": \"1782\"}, \"evidence\": \"Although the mechanism whereby autophagy and UPS function are coordinated is little understood, several regulators have emerged as important players in mediating this crosstalk, including histone deacetylase 6 (HDAC6) [50,64,75], p62/sequestosome 1 (p62) [76], and the FYVE-domain containing protein Alfy [77]; notably, these proteins have all been found to regulate or be essential for aggresome formation\", \"key\": \"38ee961f8656432e409f935d496aae7cde8d230a913610e9df3a17b84764dde7f044711ca521037b34f4af02083caa210093fae4f586c36226b1736cfd3e6207\", \"line\": 614, \"relation\": \"regulates\", \"source\": 45, \"target\": 24}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Nedelsky NB\", \"Taylor JP\", \"Todd PK\"], \"date\": \"2008-12-01\", \"first\": \"Nedelsky NB\", \"last\": \"Taylor JP\", \"name\": \"Biochimica et biophysica acta\", \"pages\": \"691-9\", \"reference\": \"18930136\", \"title\": \"Autophagy and the ubiquitin-proteasome system: collaborators in neuroprotection.\", \"type\": \"PubMed\", \"volume\": \"1782\"}, \"evidence\": \"Recent models propose that p62 and HDAC6 function analogously to facilitate autophagic degradation of proteins that display specific polyubiquitin topology.\", \"key\": \"b93ba8c2f63a4ee789311d2912df1c2bd428e3f9f9b78b6af17e6737738631ac1e313d8ebafc373836762d880620755052fdc56c8c869d684cd5452fd668717a\", \"line\": 690, \"relation\": \"increases\", \"source\": 45, \"target\": 24}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Nedelsky NB\", \"Taylor JP\", \"Todd PK\"], \"date\": \"2008-12-01\", \"first\": \"Nedelsky NB\", \"last\": \"Taylor JP\", \"name\": \"Biochimica et biophysica acta\", \"pages\": \"691-9\", \"reference\": \"18930136\", \"title\": \"Autophagy and the ubiquitin-proteasome system: collaborators in neuroprotection.\", \"type\": \"PubMed\", \"volume\": \"1782\"}, \"evidence\": \"Although the mechanism whereby autophagy and UPS function are coordinated is little understood, several regulators have emerged as important players in mediating this crosstalk, including histone deacetylase 6 (HDAC6) [50,64,75], p62/sequestosome 1 (p62) [76], and the FYVE-domain containing protein Alfy [77]; notably, these proteins have all been found to regulate or be essential for aggresome formation\", \"key\": \"19be3e1bc8ce521663d30cf9abd452a98cafab7d06371789ab1b6ed9d141b68635560a5b6b8d95c400963b94b4ebc520c7bbf45236b8737390e58c35a44fe72d\", \"line\": 615, \"relation\": \"regulates\", \"source\": 45, \"target\": 19}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Nedelsky NB\", \"Taylor JP\", \"Todd PK\"], \"date\": \"2008-12-01\", \"first\": \"Nedelsky NB\", \"last\": \"Taylor JP\", \"name\": \"Biochimica et biophysica acta\", \"pages\": \"691-9\", \"reference\": \"18930136\", \"title\": \"Autophagy and the ubiquitin-proteasome system: collaborators in neuroprotection.\", \"type\": \"PubMed\", \"volume\": \"1782\"}, \"evidence\": \"Although the mechanism whereby autophagy and UPS function are coordinated is little understood, several regulators have emerged as important players in mediating this crosstalk, including histone deacetylase 6 (HDAC6) [50,64,75], p62/sequestosome 1 (p62) [76], and the FYVE-domain containing protein Alfy [77]; notably, these proteins have all been found to regulate or be essential for aggresome formation\", \"key\": \"56524867fda3d5fdffb7473736be76ff0bdd020fd0dc918d0a22e6c59116db063a95bbb9121bbf24723a2c195203277b876d598928f34559d7b1fb62a881f4bf\", \"line\": 616, \"relation\": \"regulates\", \"source\": 45, \"target\": 3}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Nedelsky NB\", \"Taylor JP\", \"Todd PK\"], \"date\": \"2008-12-01\", \"first\": \"Nedelsky NB\", \"last\": \"Taylor JP\", \"name\": \"Biochimica et biophysica acta\", \"pages\": \"691-9\", \"reference\": \"18930136\", \"title\": \"Autophagy and the ubiquitin-proteasome system: collaborators in neuroprotection.\", \"type\": \"PubMed\", \"volume\": \"1782\"}, \"evidence\": \"HDAC6 activity appears to be important for trafficking ubiquitinated proteins and lysosomes in vitro and this has led to the suggestion that HDAC6 coordinates delivery of substrates to autophagic machinery [64,70,78].\", \"key\": \"89c8e3b11f3b5d19b5c7526202a39219f21cb1a4e394107f99653f05b6518eea2c21bb1c2d483cb90a61acf21ef4f1268da72c4356b04f27085e9f3dc8a422a3\", \"line\": 630, \"object\": {\"effect\": {\"fromLoc\": {\"name\": \"cytoplasm\", \"namespace\": \"GO\"}, \"toLoc\": {\"name\": \"lysosome\", \"namespace\": \"GO\"}}, \"modifier\": \"Translocation\"}, \"relation\": \"increases\", \"source\": 45, \"target\": 36}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Nedelsky NB\", \"Taylor JP\", \"Todd PK\"], \"date\": \"2008-12-01\", \"first\": \"Nedelsky NB\", \"last\": \"Taylor JP\", \"name\": \"Biochimica et biophysica acta\", \"pages\": \"691-9\", \"reference\": \"18930136\", \"title\": \"Autophagy and the ubiquitin-proteasome system: collaborators in neuroprotection.\", \"type\": \"PubMed\", \"volume\": \"1782\"}, \"evidence\": \"HDAC6 activity was also reported to regulate chaperone expression in response to heat shock by deacetylating Hsp90 leading to release and activation of the transcription factor HSF-1 [79].\", \"key\": \"1f7926bc02b6a3ab1ef6d22d502ae4438cb1319d9c3772850f82f5ee18b021d30208fea8c7c05e8f3c6ff7fd73b454dbb1b20fcaac548d619b4a5ddf96defc1f\", \"line\": 637, \"relation\": \"increases\", \"source\": 45, \"target\": 16}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Nedelsky NB\", \"Taylor JP\", \"Todd PK\"], \"date\": \"2008-12-01\", \"first\": \"Nedelsky NB\", \"last\": \"Taylor JP\", \"name\": \"Biochimica et biophysica acta\", \"pages\": \"691-9\", \"reference\": \"18930136\", \"title\": \"Autophagy and the ubiquitin-proteasome system: collaborators in neuroprotection.\", \"type\": \"PubMed\", \"volume\": \"1782\"}, \"evidence\": \"HDAC6 activity was also reported to regulate chaperone expression in response to heat shock by deacetylating Hsp90 leading to release and activation of the transcription factor HSF-1 [79].\", \"key\": \"940bf58d36d3455b3c9842599df6df89d0e52daea32b6b42c2a5e0945ee2657d88841297ee394d85e7d73d3fe065bb630ee4ffb3472781a85417bf5a600dcc4f\", \"line\": 638, \"relation\": \"regulates\", \"source\": 45, \"target\": 25}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Nedelsky NB\", \"Taylor JP\", \"Todd PK\"], \"date\": \"2008-12-01\", \"first\": \"Nedelsky NB\", \"last\": \"Taylor JP\", \"name\": \"Biochimica et biophysica acta\", \"pages\": \"691-9\", \"reference\": \"18930136\", \"title\": \"Autophagy and the ubiquitin-proteasome system: collaborators in neuroprotection.\", \"type\": \"PubMed\", \"volume\": \"1782\"}, \"evidence\": \"Recent models propose that p62 and HDAC6 function analogously to facilitate autophagic degradation of proteins that display specific polyubiquitin topology.\", \"key\": \"a81f61c47444c7b23aee62e634f88084c1878067c407cfcc687c0f45d1ad059ca732df20f8e12f0f09f25ccd0f270a5d27914a31236b1694133f0f076e34bf58\", \"line\": 691, \"relation\": \"increases\", \"source\": 45, \"target\": 23}, {\"annotations\": {\"Confidence\": {\"High\": true}, \"MeSHAnatomy\": {\"Endoplasmic Reticulum\": true}}, \"citation\": {\"authors\": [\"Alegre-Abarrategui J\", \"Bengoa-Vergniory N\", \"Roberts RF\", \"Wade-Martins R\"], \"date\": \"2017-12-01\", \"first\": \"Bengoa-Vergniory N\", \"last\": \"Alegre-Abarrategui J\", \"name\": \"Acta neuropathologica\", \"pages\": \"819-838\", \"reference\": \"28803412\", \"title\": \"Alpha-synuclein oligomers: a new hope.\", \"type\": \"PubMed\", \"volume\": \"134\"}, \"evidence\": \"Finally, inhibition of histone deacetylase 6 (HDAC6), which was previously shown to be involved in the response to cytotoxic ubiquitinated aggregates, increased the oligomeric content in vitro, while overexpression of HDAC6 produced the opposite effect [36]. \", \"key\": \"9b065fb16f2e4823ecddad4a0dc3ba3d709aae0579f230bf40f55f6c1e744233505a8afb755b38a73fd891dd216a981f099118622253590429385923b9b53f41\", \"line\": 120, \"relation\": \"decreases\", \"source\": 45, \"target\": 7}, {\"annotations\": {\"Confidence\": {\"Medium\": true}, \"MeSHAnatomy\": {\"Muscles\": true, \"Neurons\": true}, \"Species\": {\"7227\": true}}, \"citation\": {\"authors\": [\"Jin S\", \"Wu J\", \"Xiong Y\", \"Xu Z\", \"Zhang YQ\", \"Zhao K\"], \"date\": \"2013-03-19\", \"first\": \"Xiong Y\", \"last\": \"Zhang YQ\", \"name\": \"Proceedings of the National Academy of Sciences of the United States of America\", \"pages\": \"4604-9\", \"reference\": \"23487739\", \"title\": \"HDAC6 mutations rescue human tau-induced microtubule defects in Drosophila.\", \"type\": \"PubMed\", \"volume\": \"110\"}, \"evidence\": \"Overexpressed tau was hyperphosphorylated and resulted in decreased MT density and greater fragmentation. Using genetic screen, a histone deacetylase 6 (HDAC6) null mutation rescued tau-induced MT defects in both muscles and neurons. Genetic and pharmacological inhibition of the tubulin-specific deacetylase activity of HDAC6 indicates that the rescue effect may be mediated by increased MT acetylation.\", \"key\": \"f7dfb7d073e40b91003557a5b264d151d5735b8033dc5455de763873d663650260532f211073637c16ecca472215e8ebd915739b43d1b754044be52edfb891be\", \"line\": 4036, \"relation\": \"decreases\", \"source\": 45, \"target\": 4}, {\"citation\": {\"authors\": [\"Monahan Z\", \"Pandey UB\", \"Shewmaker F\"], \"date\": \"2016-10-15\", \"first\": \"Monahan Z\", \"last\": \"Pandey UB\", \"name\": \"Brain research\", \"pages\": \"189-200\", \"reference\": \"27181519\", \"title\": \"Stress granules at the intersection of autophagy and ALS.\", \"type\": \"PubMed\", \"volume\": \"1649\"}, \"evidence\": \"The final step in autophagy is the fusion of the autophagosome with the lysosome. This organelle maturation appears to be mediated by the cytosolic protein HDAC6, which contains ubiquitin-binding and dynein-interacting domains (Kawaguchi et al., 2003; Lee et al., 2010).\", \"key\": \"36ce8b7ae0bd03920f1bf655f59fefdb6c50fa238bd527fe09754c020fbbe26850092f1183eb07b89920b7f0a37c5e6c48312070d997e7b5f1456a717cc3db31\", \"line\": 458, \"relation\": \"increases\", \"source\": 45, \"target\": 17}, {\"citation\": {\"authors\": [\"Monahan Z\", \"Pandey UB\", \"Shewmaker F\"], \"date\": \"2016-10-15\", \"first\": \"Monahan Z\", \"last\": \"Pandey UB\", \"name\": \"Brain research\", \"pages\": \"189-200\", \"reference\": \"27181519\", \"title\": \"Stress granules at the intersection of autophagy and ALS.\", \"type\": \"PubMed\", \"volume\": \"1649\"}, \"evidence\": \"Moreover, genetic or pharmacological inhibition of HDAC6 impairs SG formation, suggesting that HDAC6 is essential for the assembly of SGs (Kwon et al., 2007).\", \"key\": \"38817d9d5a1494cb109028c247d1239ee1e2de4b94878846d0ac016be7d675ea68b3fb39972aed1dfe158d2de47a098c8a5988b3d687581ccd2ff39ef78ddcc0\", \"line\": 469, \"relation\": \"increases\", \"source\": 45, \"target\": 22}, {\"citation\": {\"authors\": [\"Mandelkow E\", \"Wang Y\"], \"date\": \"2016-01-01\", \"first\": \"Wang Y\", \"last\": \"Mandelkow E\", \"name\": \"Nature reviews. Neuroscience\", \"pages\": \"5-21\", \"reference\": \"26631930\", \"title\": \"Tau in physiology and pathology.\", \"type\": \"PubMed\", \"volume\": \"17\"}, \"evidence\": \"Tau can be acetylated by the P300 acetyltransferase or by CREB-binding protein at several Lys residues in the flanking region or the repeat domain, and deacetylated at these sites by sirtuin 1 (SIRT1) and histone deacetylase 6 (HDAC6), respectively\", \"key\": \"0cd0a4240bd341c58aa3dbd0f6d72104d73227cedd919d20b50ad5777b4da933c4f377f11e64c5ba584689232e40f4842ab4727e39520b60d55bb59690ef37f3\", \"line\": 249, \"relation\": \"decreases\", \"source\": 45, \"target\": 48}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Carlomagno Y\", \"Castanedes-Casey M\", \"Chung DC\", \"Cook C\", \"DeTure M\", \"Dickson DW\", \"Dunmore J\", \"Madden BJ\", \"Petrucelli L\", \"Tong J\", \"Yue M\"], \"date\": \"2017-09-15\", \"first\": \"Carlomagno Y\", \"last\": \"Cook C\", \"name\": \"The Journal of biological chemistry\", \"pages\": \"15277-15286\", \"reference\": \"28760828\", \"title\": \"An acetylation-phosphorylation switch that regulates tau aggregation propensity and function.\", \"type\": \"PubMed\", \"volume\": \"292\"}, \"evidence\": \"Our findings indicate that several acetylation sites in tau are responsive to HDAC6 and that acetylation on Lys-321 (within a KCGS motif) is both essential for acetylation-mediated inhibition of tau aggregation in vitro and a molecular tactic for preventing phosphorylation on the downstream Ser-324 residue. Tau phosphorylation of Ser-324 (pSer-324) has not previously been evaluated in the context of tauopathy, and here we observed increased deposition of pSer-324-positive tau both in mouse models of tauopathy and in patients with Alzheimer's disease. These findings uncover a novel acetylation-phosphorylation switch at Lys-321/Ser-324 that coordinately regulates tau polymerization and function.\", \"key\": \"ad843db257243ad7004a4642c6a29fb9285f062cc11047dedc1cce8e77930138fbbaa72cb5c3732607bd883dab634da756358f04dc508224fc7d2221f676043e\", \"line\": 4156, \"relation\": \"increases\", \"source\": 45, \"target\": 49}, {\"annotations\": {\"Confidence\": {\"High\": true}}, \"citation\": {\"authors\": [\"Blair LJ\", \"Koren J 3rd\", \"Shelton LB\"], \"date\": \"2017-01-01\", \"first\": \"Shelton LB\", \"last\": \"Blair LJ\", \"name\": \"Frontiers in neuroscience\", \"pages\": \"724\", \"reference\": \"29311797\", \"title\": \"Imbalances in the Hsp90 Chaperone Machinery: Implications for Tauopathies.\", \"type\": \"PubMed\", \"volume\": \"11\"}, \"evidence\": \"While a direct role of XAP2 in tau pathogenesis has not been described, studies have shown that XAP2 is activated by histone deacetylase (HDAC) 6, which has been linked to pathogenic tau (Kekatpure et al., 2009; Cook et al., 2012; Selenica et al., 2014).\", \"key\": \"ab7f970b007d46532600cdd2625de6695567f6b643bdaa7c0c62d9b397de55723fdfb516cddd487824e637fc3d6102eed11e49bcf4136e5bdefe117438698b9d\", \"line\": 273, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"increases\", \"source\": 45, \"target\": 37}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Berkovitch SS\", \"Chattopadhyay S\", \"Haggarty SJ\", \"Huang JH\", \"Iaconelli J\", \"Karmacharya R\", \"Mazitschek R\", \"Schreiber SL\"], \"date\": \"2015-03-20\", \"first\": \"Iaconelli J\", \"last\": \"Karmacharya R\", \"name\": \"ACS chemical biology\", \"pages\": \"883-90\", \"reference\": \"25546293\", \"title\": \"HDAC6 inhibitors modulate Lys49 acetylation and membrane localization of β-catenin in human iPSC-derived neuronal cells.\", \"type\": \"PubMed\", \"volume\": \"10\"}, \"evidence\": \"The N-terminus of β-catenin has phosphorylation, ubiquitination, and acetylation sites that regulate its stability and signaling. In the absence of a Wnt signal, Ser33, Ser37, and Thr41 are constitutively phosphorylated by glycogen synthase kinase 3β (GSK3β). β-Catenin phosphorylated at these sites is recognized by β-transducin repeat-containing protein (βTrCP), which results in ubiquitination and degradation by the ubiquitin-proteasome pathway. The N-terminal regulatory domain of β-catenin also includes Ser45, a phosphorylation site for Casein Kinase 1α (CK1α) and Lys49, which is acetylated by the acetyltransferase p300/CBP-associated factor (PCAF). The relevance of Lys49 acetylation and Ser45 phosphorylation to the function of β-catenin is an active area of investigation. We find that HDAC6 inhibitors increase Lys49 acetylation and Ser45 phosphorylation but do not affect Ser33, Ser37, and Thr41 phosphorylation. Lys49 acetylation results in decreased ubiquitination of β-catenin in the presence of proteasome inhibition. While increased Lys49 acetylation does not affect total levels of β-catenin, it results in increased membrane localization of β-catenin.\", \"key\": \"ecf358a2ee9729afd61ce37ec706fc7a64fc8fcff2eb02972c81fe2f729657698b88aef321efab63f1d9381e2b1bc1898a1623a9910b038a78c892e5486ec26b\", \"line\": 3693, \"relation\": \"causesNoChange\", \"source\": 45, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 40}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Berkovitch SS\", \"Chattopadhyay S\", \"Haggarty SJ\", \"Huang JH\", \"Iaconelli J\", \"Karmacharya R\", \"Mazitschek R\", \"Schreiber SL\"], \"date\": \"2015-03-20\", \"first\": \"Iaconelli J\", \"last\": \"Karmacharya R\", \"name\": \"ACS chemical biology\", \"pages\": \"883-90\", \"reference\": \"25546293\", \"title\": \"HDAC6 inhibitors modulate Lys49 acetylation and membrane localization of β-catenin in human iPSC-derived neuronal cells.\", \"type\": \"PubMed\", \"volume\": \"10\"}, \"evidence\": \"The N-terminus of β-catenin has phosphorylation, ubiquitination, and acetylation sites that regulate its stability and signaling. In the absence of a Wnt signal, Ser33, Ser37, and Thr41 are constitutively phosphorylated by glycogen synthase kinase 3β (GSK3β). β-Catenin phosphorylated at these sites is recognized by β-transducin repeat-containing protein (βTrCP), which results in ubiquitination and degradation by the ubiquitin-proteasome pathway. The N-terminal regulatory domain of β-catenin also includes Ser45, a phosphorylation site for Casein Kinase 1α (CK1α) and Lys49, which is acetylated by the acetyltransferase p300/CBP-associated factor (PCAF). The relevance of Lys49 acetylation and Ser45 phosphorylation to the function of β-catenin is an active area of investigation. We find that HDAC6 inhibitors increase Lys49 acetylation and Ser45 phosphorylation but do not affect Ser33, Ser37, and Thr41 phosphorylation. Lys49 acetylation results in decreased ubiquitination of β-catenin in the presence of proteasome inhibition. While increased Lys49 acetylation does not affect total levels of β-catenin, it results in increased membrane localization of β-catenin.\", \"key\": \"63643d54d0024d5db31c24981336dd8b7ceb6255659dc94f3b119c6c8e464650d2d717e3ac47e4282345e0e62181bc169ea019e5976e0b2a8d45aac66008b19c\", \"line\": 3694, \"relation\": \"causesNoChange\", \"source\": 45, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 41}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Berkovitch SS\", \"Chattopadhyay S\", \"Haggarty SJ\", \"Huang JH\", \"Iaconelli J\", \"Karmacharya R\", \"Mazitschek R\", \"Schreiber SL\"], \"date\": \"2015-03-20\", \"first\": \"Iaconelli J\", \"last\": \"Karmacharya R\", \"name\": \"ACS chemical biology\", \"pages\": \"883-90\", \"reference\": \"25546293\", \"title\": \"HDAC6 inhibitors modulate Lys49 acetylation and membrane localization of β-catenin in human iPSC-derived neuronal cells.\", \"type\": \"PubMed\", \"volume\": \"10\"}, \"evidence\": \"The N-terminus of β-catenin has phosphorylation, ubiquitination, and acetylation sites that regulate its stability and signaling. In the absence of a Wnt signal, Ser33, Ser37, and Thr41 are constitutively phosphorylated by glycogen synthase kinase 3β (GSK3β). β-Catenin phosphorylated at these sites is recognized by β-transducin repeat-containing protein (βTrCP), which results in ubiquitination and degradation by the ubiquitin-proteasome pathway. The N-terminal regulatory domain of β-catenin also includes Ser45, a phosphorylation site for Casein Kinase 1α (CK1α) and Lys49, which is acetylated by the acetyltransferase p300/CBP-associated factor (PCAF). The relevance of Lys49 acetylation and Ser45 phosphorylation to the function of β-catenin is an active area of investigation. We find that HDAC6 inhibitors increase Lys49 acetylation and Ser45 phosphorylation but do not affect Ser33, Ser37, and Thr41 phosphorylation. Lys49 acetylation results in decreased ubiquitination of β-catenin in the presence of proteasome inhibition. While increased Lys49 acetylation does not affect total levels of β-catenin, it results in increased membrane localization of β-catenin.\", \"key\": \"9c2e29a4e7c506c5f7c97bb1a53c8dbf8b3fec0371c6e5d4e12037983e359e60d525511dde20e2aff3a88411a3bba147cc513e1a7d1200c813639be4e888d0e3\", \"line\": 3691, \"relation\": \"decreases\", \"source\": 45, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 39}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Carlomagno Y\", \"Castanedes-Casey M\", \"Chung DC\", \"Cook C\", \"DeTure M\", \"Dickson DW\", \"Dunmore J\", \"Madden BJ\", \"Petrucelli L\", \"Tong J\", \"Yue M\"], \"date\": \"2017-09-15\", \"first\": \"Carlomagno Y\", \"last\": \"Cook C\", \"name\": \"The Journal of biological chemistry\", \"pages\": \"15277-15286\", \"reference\": \"28760828\", \"title\": \"An acetylation-phosphorylation switch that regulates tau aggregation propensity and function.\", \"type\": \"PubMed\", \"volume\": \"292\"}, \"evidence\": \"HDAC6 inhibition leads to a significant reduction in tau levels as detected by the human tau-specific antibody E1 (Fig. 6 (a and c) and supplemental Fig. S6). We also observed a striking decrease in phosphorylation at Ser-324, which was statistically significant even when normalizing to E1 to control for the reduction in tau levels (Fig. 6 (a and b) and supplemental Fig. S6).\", \"key\": \"a1df6254531792594e0fe11f260a607c52ca6a0a0bd7c21573d08b1f56645f88bb9ce1763f80048079ffe3082c233e7a6f073163f3d355d71a9a5ecde8a2f78f\", \"line\": 4165, \"relation\": \"increases\", \"source\": 45, \"target\": 46}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Carlomagno Y\", \"Cook C\", \"Petrucelli L\", \"Stankowski JN\", \"Stetler C\"], \"date\": \"2014-01-01\", \"first\": \"Cook C\", \"last\": \"Petrucelli L\", \"name\": \"Alzheimer's research &amp; therapy\", \"pages\": \"29\", \"reference\": \"25031639\", \"title\": \"Acetylation: a new key to unlock tau's role in neurodegeneration.\", \"type\": \"PubMed\", \"volume\": \"6\"}, \"evidence\": \"We demonstrate that elevated HDAC6 activity increases phosphorylation of tau at the 12E8 epitope (pS262/356), a phospho-epitope present within the KXGS motifs of tau’s microtubule-binding domain. The phosphorylation of KXGS motifs within tau by the kinase Par-1/MARK2 is required for tau proteotoxicity in Drosophila [29], observed at very early stages of NFT formation in AD brain [30], and appears to prime tau for subsequent phosphorylation events\", \"key\": \"ec192ef0a5a8448e3884bd9f44401f8eb34ef6aec0bd31c2fda28941e8d243925937eb595f5c79247042711d829cad0f604012455860a6f81bf58c9acc2de022\", \"line\": 3278, \"relation\": \"increases\", \"source\": 45, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 50}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Carlomagno Y\", \"Cook C\", \"Petrucelli L\", \"Stankowski JN\", \"Stetler C\"], \"date\": \"2014-01-01\", \"first\": \"Cook C\", \"last\": \"Petrucelli L\", \"name\": \"Alzheimer's research &amp; therapy\", \"pages\": \"29\", \"reference\": \"25031639\", \"title\": \"Acetylation: a new key to unlock tau's role in neurodegeneration.\", \"type\": \"PubMed\", \"volume\": \"6\"}, \"evidence\": \"We demonstrate that elevated HDAC6 activity increases phosphorylation of tau at the 12E8 epitope (pS262/356), a phospho-epitope present within the KXGS motifs of tau’s microtubule-binding domain. The phosphorylation of KXGS motifs within tau by the kinase Par-1/MARK2 is required for tau proteotoxicity in Drosophila [29], observed at very early stages of NFT formation in AD brain [30], and appears to prime tau for subsequent phosphorylation events\", \"key\": \"9ad88d234f3be2eafd7a9890f0a704f9940e4c5383c43a3abe10edf351b10c897969c70d6b8f6b46759fe7aaf821acb11c46cb6bc2a77de84bc8a0a51cd6394f\", \"line\": 3279, \"relation\": \"increases\", \"source\": 45, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 52}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Carlomagno Y\", \"Castanedes-Casey M\", \"Chung DC\", \"Cook C\", \"DeTure M\", \"Dickson DW\", \"Dunmore J\", \"Madden BJ\", \"Petrucelli L\", \"Tong J\", \"Yue M\"], \"date\": \"2017-09-15\", \"first\": \"Carlomagno Y\", \"last\": \"Cook C\", \"name\": \"The Journal of biological chemistry\", \"pages\": \"15277-15286\", \"reference\": \"28760828\", \"title\": \"An acetylation-phosphorylation switch that regulates tau aggregation propensity and function.\", \"type\": \"PubMed\", \"volume\": \"292\"}, \"evidence\": \"HDAC6 inhibition leads to a significant reduction in tau levels as detected by the human tau-specific antibody E1 (Fig. 6 (a and c) and supplemental Fig. S6). We also observed a striking decrease in phosphorylation at Ser-324, which was statistically significant even when normalizing to E1 to control for the reduction in tau levels (Fig. 6 (a and b) and supplemental Fig. S6).\", \"key\": \"b5f3f67fa08232fd5306e5b9b5270cb6cf9ab3c74cc39364bec7f22feef12f2d9d76ad396f4afc0e73631f0e6ea15d89acafd4a27c53302bfe146137263c8dc7\", \"line\": 4166, \"relation\": \"increases\", \"source\": 45, \"target\": 51}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Berkovitch SS\", \"Chattopadhyay S\", \"Haggarty SJ\", \"Huang JH\", \"Iaconelli J\", \"Karmacharya R\", \"Mazitschek R\", \"Schreiber SL\"], \"date\": \"2015-03-20\", \"first\": \"Iaconelli J\", \"last\": \"Karmacharya R\", \"name\": \"ACS chemical biology\", \"pages\": \"883-90\", \"reference\": \"25546293\", \"title\": \"HDAC6 inhibitors modulate Lys49 acetylation and membrane localization of β-catenin in human iPSC-derived neuronal cells.\", \"type\": \"PubMed\", \"volume\": \"10\"}, \"evidence\": \"The N-terminus of β-catenin has phosphorylation, ubiquitination, and acetylation sites that regulate its stability and signaling. In the absence of a Wnt signal, Ser33, Ser37, and Thr41 are constitutively phosphorylated by glycogen synthase kinase 3β (GSK3β). β-Catenin phosphorylated at these sites is recognized by β-transducin repeat-containing protein (βTrCP), which results in ubiquitination and degradation by the ubiquitin-proteasome pathway. The N-terminal regulatory domain of β-catenin also includes Ser45, a phosphorylation site for Casein Kinase 1α (CK1α) and Lys49, which is acetylated by the acetyltransferase p300/CBP-associated factor (PCAF). The relevance of Lys49 acetylation and Ser45 phosphorylation to the function of β-catenin is an active area of investigation. We find that HDAC6 inhibitors increase Lys49 acetylation and Ser45 phosphorylation but do not affect Ser33, Ser37, and Thr41 phosphorylation. Lys49 acetylation results in decreased ubiquitination of β-catenin in the presence of proteasome inhibition. While increased Lys49 acetylation does not affect total levels of β-catenin, it results in increased membrane localization of β-catenin.\", \"key\": \"5ad2a3f018a0eb40b7d65468f2a9f1af19913d30e80d0d8f744e8f59a5ae581c0c1fde2b6c929efdb3bf3a0bb6dff04da6c9f46ca4117344365e8616a41e8c5e\", \"line\": 3692, \"relation\": \"decreases\", \"source\": 45, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 42}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Berkovitch SS\", \"Chattopadhyay S\", \"Haggarty SJ\", \"Huang JH\", \"Iaconelli J\", \"Karmacharya R\", \"Mazitschek R\", \"Schreiber SL\"], \"date\": \"2015-03-20\", \"first\": \"Iaconelli J\", \"last\": \"Karmacharya R\", \"name\": \"ACS chemical biology\", \"pages\": \"883-90\", \"reference\": \"25546293\", \"title\": \"HDAC6 inhibitors modulate Lys49 acetylation and membrane localization of β-catenin in human iPSC-derived neuronal cells.\", \"type\": \"PubMed\", \"volume\": \"10\"}, \"evidence\": \"The N-terminus of β-catenin has phosphorylation, ubiquitination, and acetylation sites that regulate its stability and signaling. In the absence of a Wnt signal, Ser33, Ser37, and Thr41 are constitutively phosphorylated by glycogen synthase kinase 3β (GSK3β). β-Catenin phosphorylated at these sites is recognized by β-transducin repeat-containing protein (βTrCP), which results in ubiquitination and degradation by the ubiquitin-proteasome pathway. The N-terminal regulatory domain of β-catenin also includes Ser45, a phosphorylation site for Casein Kinase 1α (CK1α) and Lys49, which is acetylated by the acetyltransferase p300/CBP-associated factor (PCAF). The relevance of Lys49 acetylation and Ser45 phosphorylation to the function of β-catenin is an active area of investigation. We find that HDAC6 inhibitors increase Lys49 acetylation and Ser45 phosphorylation but do not affect Ser33, Ser37, and Thr41 phosphorylation. Lys49 acetylation results in decreased ubiquitination of β-catenin in the presence of proteasome inhibition. While increased Lys49 acetylation does not affect total levels of β-catenin, it results in increased membrane localization of β-catenin.\", \"key\": \"7526e46dde29fa5480e10fc36ae13c8719c81383ea97015caa6b3ad70573f763d38f85beb4e3fd24169cb2f71aa582f793b38d4708b4f760470918748ab47938\", \"line\": 3695, \"relation\": \"causesNoChange\", \"source\": 45, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 43}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Huang L\", \"Liu C\", \"Qin C\", \"Sheng SL\", \"Sui XL\", \"Tao JJ\", \"Wu J\", \"Xu YF\", \"Yao ZG\", \"Zhang L\", \"Zhu H\"], \"date\": \"2014-01-01\", \"first\": \"Zhang L\", \"last\": \"Qin C\", \"name\": \"Journal of Alzheimer's disease : JAD\", \"pages\": \"1193-205\", \"reference\": \"24844691\", \"title\": \"Tubastatin A/ACY-1215 improves cognition in Alzheimer's disease transgenic mice.\", \"type\": \"PubMed\", \"volume\": \"41\"}, \"evidence\": \"In the present study, we tested the potential of two selective HDAC6 inhibitors, tubastatin A and ACY-1215, to rescue cognitive deficits in a mouse model of AD. We found that both tubastatin A and ACY-1215 alleviated behavioral deficits, altered amyloid-β (Aβ) load, and reduced tau hyperphosphorylation in AD mice without obvious adverse effects. Our data suggested that tubastatin A and ACY-1215 not only promoted tubulin acetylation, but also reduced production and facilitated autophagic clearance of Aβ and hyperphosphorylated tau.\", \"key\": \"45c6d221208aa12da9ee0915a5baf4f1a41f3d242089bbe1d115c0b899cfd85b9c22df11b4713de8d8b6a253171ccf1d6f4e27d34d3c3e1d51e3f0726c409d32\", \"line\": 3647, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"directlyDecreases\", \"source\": 13, \"target\": 45}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Huang L\", \"Liu C\", \"Qin C\", \"Sheng SL\", \"Sui XL\", \"Tao JJ\", \"Wu J\", \"Xu YF\", \"Yao ZG\", \"Zhang L\", \"Zhu H\"], \"date\": \"2014-01-01\", \"first\": \"Zhang L\", \"last\": \"Qin C\", \"name\": \"Journal of Alzheimer's disease : JAD\", \"pages\": \"1193-205\", \"reference\": \"24844691\", \"title\": \"Tubastatin A/ACY-1215 improves cognition in Alzheimer's disease transgenic mice.\", \"type\": \"PubMed\", \"volume\": \"41\"}, \"evidence\": \"In the present study, we tested the potential of two selective HDAC6 inhibitors, tubastatin A and ACY-1215, to rescue cognitive deficits in a mouse model of AD. We found that both tubastatin A and ACY-1215 alleviated behavioral deficits, altered amyloid-β (Aβ) load, and reduced tau hyperphosphorylation in AD mice without obvious adverse effects. Our data suggested that tubastatin A and ACY-1215 not only promoted tubulin acetylation, but also reduced production and facilitated autophagic clearance of Aβ and hyperphosphorylated tau.\", \"key\": \"625cc21c5f217d1d03f3dac923514e6f53aaad97501ff8006744f5704cb48add82b02a8861e6be1330f8bb883a78e6b336ef87f8446470da09ae0a0a23921a47\", \"line\": 3646, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"directlyDecreases\", \"source\": 12, \"target\": 45}, {\"annotations\": {\"Confidence\": {\"High\": true}, \"IC50\": {\"65 nM\": true}, \"Ki\": {\"110 nM\": true}}, \"citation\": {\"authors\": [\"Beckers T\", \"Burkhardt C\", \"Ciossek T\", \"Dehmel F\", \"Fettis K\", \"Julius H\", \"Maier T\", \"Stengel T\", \"Weinbrenner S\", \"Wieland H\"], \"date\": \"2008-07-10\", \"first\": \"Dehmel F\", \"last\": \"Beckers T\", \"name\": \"Journal of medicinal chemistry\", \"pages\": \"3985-4001\", \"reference\": \"18558669\", \"title\": \"Trithiocarbonates as a novel class of HDAC inhibitors: SAR studies, isoenzyme selectivity, and pharmacological profiles.\", \"type\": \"PubMed\", \"volume\": \"51\"}, \"evidence\": \"Highly potent, substrate competitive HDAC6 selective inhibitors were identified (12ac:IC 50 = 65 nM and K i = 110 nM). Trithiocarbonate analogues with an aminoquinoline-substituted pyridinyl-thienoacetyl cap demonstrate a cytotoxicity profile and potency comparable to that of suberoylanilide hydroxamic acid (SAHA) as an approved cancer drug.\", \"key\": \"382c980f11cd72afbfa20087b26b8b0238fa8eb4c6797925b162e8e1a5be04b1e1cc4c4121b063e3e74920fc1f13726c8ec15e84ad13ec3599f521473dfc46e9\", \"line\": 3725, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"directlyDecreases\", \"source\": 9, \"target\": 45}, {\"annotations\": {\"Cell_Line\": {\"N2a\": true, \"SH-SY5Y\": true}, \"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Fan SJ\", \"Huang FI\", \"Liou JP\", \"Yang CR\"], \"date\": \"2018-05-29\", \"first\": \"Fan SJ\", \"last\": \"Yang CR\", \"name\": \"Cell death &amp; disease\", \"pages\": \"655\", \"reference\": \"29844403\", \"title\": \"The novel histone de acetylase 6 inhibitor, MPT0G211, ameliorates tau phosphorylation and cognitive deficits in an Alzheimer's disease model.\", \"type\": \"PubMed\", \"volume\": \"9\"}, \"evidence\": \"This study aimed to investigate the protective effects and mechanism of the novel HDAC6 inhibitor, MPT0G211, using an AD model. Our results indicated that MPT0G211 significantly reduced tau phosphorylation and aggregation, the processes highly correlated with the formation of NFTs. This HDAC6 inhibitory activity resulted in an increase in acetylated Hsp90, which decreased Hsp90 and HDAC6 binding, causing ubiquitination of phosphorylated tau proteins. In addition, a significant increase of phospho-glycogen synthase kinase-3β (phospho-GSK3β) on Ser9 (the inactive form) through Akt phosphorylation was associated with the inhibition of phospho-tau Ser396 in response to MPT0G211 treatment.\", \"key\": \"bdf80756c972d79265301286a4e51b963a7c9143950165c945c31c97536dad9958fad4020d77fe4779ac394cfc34acf4f088d3e45b96a9b5bb009e77788f074f\", \"line\": 4200, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"directlyDecreases\", \"source\": 10, \"target\": 45}, {\"key\": \"72f5a79a0544d96818d72cda97ec26df76e9eb31ff0a94ceb2c3343e7b1e4120d64d60802ae4b4d7bdcca138073fadf5b82af444ad2ec846ad0e337408da7702\", \"relation\": \"hasComponent\", \"source\": 28, \"target\": 45}, {\"key\": \"72f5a79a0544d96818d72cda97ec26df76e9eb31ff0a94ceb2c3343e7b1e4120d64d60802ae4b4d7bdcca138073fadf5b82af444ad2ec846ad0e337408da7702\", \"relation\": \"hasComponent\", \"source\": 28, \"target\": 32}, {\"key\": \"df6a4d6319f7c849703dc37be903df7cec7813d723af3ad96268072d98c8f88d2d1a3fca6d653ff1b708a294b741947c68148bfe9e0a1838a198579e6289ad00\", \"relation\": \"hasComponent\", \"source\": 28, \"target\": 34}, {\"key\": \"6ae28b3f01a6ca666613d588168ed1e141f6212a84fb29f938445ba990104385e299d05bd4f33443cf3736a0305e8bb0741b9f3540a27e211820cc2311851e65\", \"relation\": \"hasComponent\", \"source\": 26, \"target\": 45}, {\"key\": \"7d7d4ae30f9b1ebe1276a0fc99d650d1466b5423f4c591e141d67dce1055841470a836c4c8d573d99fa148a9db5bec7ad6068d6336329156339b917da710bf22\", \"relation\": \"hasComponent\", \"source\": 26, \"target\": 31}, {\"key\": \"6ae28b3f01a6ca666613d588168ed1e141f6212a84fb29f938445ba990104385e299d05bd4f33443cf3736a0305e8bb0741b9f3540a27e211820cc2311851e65\", \"relation\": \"hasComponent\", \"source\": 26, \"target\": 32}, {\"citation\": {\"authors\": [\"Monahan Z\", \"Pandey UB\", \"Shewmaker F\"], \"date\": \"2016-10-15\", \"first\": \"Monahan Z\", \"last\": \"Pandey UB\", \"name\": \"Brain research\", \"pages\": \"189-200\", \"reference\": \"27181519\", \"title\": \"Stress granules at the intersection of autophagy and ALS.\", \"type\": \"PubMed\", \"volume\": \"1649\"}, \"evidence\": \"Interestingly, HDAC6 physically interacts with G3BP, a known component of SGs, and is sequestered into SGs under different stress conditions\", \"key\": \"1c8dd1b0c195ffc0d963901e4279c7ed33c4ebb30eae95ca926884fedf5de6d86477ee8438fe33a166511d6ef786400355e895611bd752feb975ab2f29c2d591\", \"line\": 465, \"object\": {\"location\": {\"name\": \"cytoplasmic stress granule\", \"namespace\": \"GO\"}}, \"relation\": \"increases\", \"source\": 21, \"target\": 45}, {\"key\": \"9aead9f4e019ff0149703e2d7c5af3fdc926a0a1456404596a0d540443a1e11e3861f694e5c95cf6db1ab2e260feb6f343befdb886fc4694ef1fffa7cac4ac2a\", \"relation\": \"hasComponent\", \"source\": 27, \"target\": 45}, {\"citation\": {\"authors\": [\"Monahan Z\", \"Pandey UB\", \"Shewmaker F\"], \"date\": \"2016-10-15\", \"first\": \"Monahan Z\", \"last\": \"Pandey UB\", \"name\": \"Brain research\", \"pages\": \"189-200\", \"reference\": \"27181519\", \"title\": \"Stress granules at the intersection of autophagy and ALS.\", \"type\": \"PubMed\", \"volume\": \"1649\"}, \"evidence\": \"Recent studies have shown that the autophagy-adapter p62 localizes with HDAC6 and controls its deacetylase activity (Yan et al., 2013)\", \"key\": \"2da2e058a0a251474bc85dad56531d88b953cf58b6b7dad199e3cf52c1d7f601b2c7509166f41ea59fe1f8ff17cd4a591b9bcddf8ad52c90eba80a11a81d2836\", \"line\": 476, \"object\": {\"effect\": {\"name\": \"protein deacetylation\", \"namespace\": \"GO\"}, \"modifier\": \"Activity\"}, \"relation\": \"regulates\", \"source\": 27, \"target\": 45}, {\"key\": \"9aead9f4e019ff0149703e2d7c5af3fdc926a0a1456404596a0d540443a1e11e3861f694e5c95cf6db1ab2e260feb6f343befdb886fc4694ef1fffa7cac4ac2a\", \"relation\": \"hasComponent\", \"source\": 27, \"target\": 32}, {\"key\": \"9bd44a2316e805f6c4b3dff8432b69220b99048428dc5e17e6cd520207baa410d5ebfc4dc86bd86af44e7ef8bc18feeb3dd62eb88b11e2f1724e018974cfd76a\", \"relation\": \"hasComponent\", \"source\": 27, \"target\": 33}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Babić Leko M\", \"Bažadona D\", \"Buée L\", \"Delalle I\", \"Di Giovanni G\", \"Harrington C\", \"Hof PR\", \"Jovanov-Milošević N\", \"Wischik C\", \"Wray S\", \"de Silva R\", \"Šimić G\"], \"date\": \"2016-01-06\", \"first\": \"Šimić G\", \"last\": \"Hof PR\", \"name\": \"Biomolecules\", \"pages\": \"6\", \"reference\": \"26751493\", \"title\": \"Tau Protein Hyperphosphorylation and Aggregation in Alzheimer's Disease and Other Tauopathies, and Possible Neuroprotective Strategies.\", \"type\": \"PubMed\", \"volume\": \"6\"}, \"evidence\": \"An additional “knot” of tau being entangled in epigenetic landscape of neurodegeneration comes from the finding that by acting as a HDAC6 inhibitor, tau is being indirectly involved in both (dys)regulation of transcriptional activity and impairment of autophagic clearance by the ubiquitin proteasome system [81,82].\", \"key\": \"f4fbd21c80321656d876bbc30fe9945997c452a41335e60f08dd30c4837247ae0773920bb80405cb1162ccc434d71e8502135109b4fae3dafc67f5ab8dd9aa06\", \"line\": 153, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"decreases\", \"source\": 46, \"target\": 45}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"authors\": [\"Jin J\", \"Liu Y\", \"Wang S\", \"Xu Y\", \"Ye X\", \"Yu L\", \"Zhu X\"], \"date\": \"2017-10-01\", \"first\": \"Zhu X\", \"last\": \"Xu Y\", \"name\": \"Aging cell\", \"pages\": \"1073-1082\", \"reference\": \"28771976\", \"title\": \"HDAC3 negatively regulates spatial memory in a mouse model of Alzheimer's disease.\", \"type\": \"PubMed\", \"volume\": \"16\"}, \"evidence\": \"RGFP966, a selective HDAC3 inhibitor, has been shown to affect sensory cortical plasticity and memory formation (Bieszczad et al., 2015).\", \"key\": \"33f293508013179e36b3cac626a1320a9dccfdd688a2cfd53607831246e8a39ead0767586966be3089038aba9413a8a875712a2766a2796de6e6afde79276c9d\", \"line\": 4187, \"object\": {\"modifier\": \"Activity\"}, \"relation\": \"directlyDecreases\", \"source\": 14, \"target\": 44}, {\"annotations\": {\"Anatomy\": {\"hippocampal formation\": true}, \"Confidence\": {\"Medium\": true}, \"Research_Model\": {\"APPPS1 mice\": true}, \"Species\": {\"10090\": true}}, \"citation\": {\"authors\": [\"Jin J\", \"Liu Y\", \"Wang S\", \"Xu Y\", \"Ye X\", \"Yu L\", \"Zhu X\"], \"date\": \"2017-10-01\", \"first\": \"Zhu X\", \"last\": \"Xu Y\", \"name\": \"Aging cell\", \"pages\": \"1073-1082\", \"reference\": \"28771976\", \"title\": \"HDAC3 negatively regulates spatial memory in a mouse model of Alzheimer's disease.\", \"type\": \"PubMed\", \"volume\": \"16\"}, \"evidence\": \"HDAC3 overexpression in the hippocampus increases Aß levels, activates microglia, and decreases dendritic spine density in 6-month-old APP/PS1 mice.\", \"key\": \"dc668c5ba18ef2a7e16b2f1369f885e2b012f1710ca9b361ec5763dc5342b56429d1f4653e9c748938c21f7fc58119050d1a7cf4dda8d568ecc20c182f393120\", \"line\": 4176, \"relation\": \"increases\", \"source\": 44, \"target\": 0}, {\"annotations\": {\"Anatomy\": {\"hippocampal formation\": true}, \"Confidence\": {\"Medium\": true}, \"Research_Model\": {\"APPPS1 mice\": true}, \"Species\": {\"10090\": true}}, \"citation\": {\"authors\": [\"Jin J\", \"Liu Y\", \"Wang S\", \"Xu Y\", \"Ye X\", \"Yu L\", \"Zhu X\"], \"date\": \"2017-10-01\", \"first\": \"Zhu X\", \"last\": \"Xu Y\", \"name\": \"Aging cell\", \"pages\": \"1073-1082\", \"reference\": \"28771976\", \"title\": \"HDAC3 negatively regulates spatial memory in a mouse model of Alzheimer's disease.\", \"type\": \"PubMed\", \"volume\": \"16\"}, \"evidence\": \"HDAC3 overexpression in the hippocampus increases Aß levels, activates microglia, and decreases dendritic spine density in 6-month-old APP/PS1 mice.\", \"key\": \"1ba6f74776bf83abfdc5d50fc8a8ab23471cac1d003c4434cf7d0aca383d5371a5006b3bf181b4ae44c146696dc607e9b914d1104ff9a3e8cdaf3d31089e7d93\", \"line\": 4177, \"relation\": \"increases\", \"source\": 44, \"target\": 18}, {\"annotations\": {\"Anatomy\": {\"hippocampal formation\": true}, \"Confidence\": {\"Medium\": true}, \"Research_Model\": {\"APPPS1 mice\": true}, \"Species\": {\"10090\": true}}, \"citation\": {\"authors\": [\"Jin J\", \"Liu Y\", \"Wang S\", \"Xu Y\", \"Ye X\", \"Yu L\", \"Zhu X\"], \"date\": \"2017-10-01\", \"first\": \"Zhu X\", \"last\": \"Xu Y\", \"name\": \"Aging cell\", \"pages\": \"1073-1082\", \"reference\": \"28771976\", \"title\": \"HDAC3 negatively regulates spatial memory in a mouse model of Alzheimer's disease.\", \"type\": \"PubMed\", \"volume\": \"16\"}, \"evidence\": \"HDAC3 overexpression in the hippocampus increases Aß levels, activates microglia, and decreases dendritic spine density in 6-month-old APP/PS1 mice.\", \"key\": \"6e61648fe7185469dd667699523a5fc4042bbacafc08f52eb4ed11dbb86c00e145fa73b13729097ee6782d97edae869ab0687ed7a335cc642a200b901a56b55a\", \"line\": 4178, \"relation\": \"decreases\", \"source\": 44, \"target\": 2}, {\"key\": \"fe93ded75ed8d4e455e62ecb78ec5d2dc1a6fb2ee44ffee156b3b8da8936a393121374afadf49aa455ed7c5f0056fa23194995b95d12b39ae15fb727537e13b1\", \"relation\": \"hasVariant\", \"source\": 35, \"target\": 36}, {\"key\": \"9f569d1191afb14f5e5a5c583118b5a066a513cd06ceb12c3e167840993ae4c26227d5d44b9694a92f9f8f0966e7daca9b061a9272c3a8fc77a1720171c7e501\", \"relation\": \"hasVariant\", \"source\": 38, \"target\": 40}], \"multigraph\": true, \"nodes\": [{\"bel\": \"a(CHEBI:\\\"amyloid-beta\\\")\", \"function\": \"Abundance\", \"id\": \"a9487b29439483c6840095610dd3e496e3ca9cccc3a8d71e0385a39b570f3c0a26965ac06d83265e1c08deaae7998272d4f4c21ec0534f8be3fd8e4e57dbb3a2\", \"name\": \"amyloid-beta\", \"namespace\": \"CHEBI\"}, {\"bel\": \"a(CHEBI:\\\"trichostatin A\\\")\", \"function\": \"Abundance\", \"id\": \"8c8269fc63126d8bacd63c3b16d5a192e864dcb954f7c1ba24df2c409bb1e0c622f0e213d1c88064ff72a4e30c866f5e7716318547299bc0588a4c568c6058dd\", \"name\": \"trichostatin A\", \"namespace\": \"CHEBI\"}, {\"bel\": \"a(GO:\\\"dendritic spine\\\")\", \"function\": \"Abundance\", \"id\": \"50b68b3067609df23d68ed8be7713e3e896c5d94c5c9422cbfdb7c7a5856c1c67afb00619691ad2f190eda88738c1b51bccc9e52a506f6209908caddf176342b\", \"name\": \"dendritic spine\", \"namespace\": \"GO\"}, {\"bel\": \"a(GO:aggresome)\", \"function\": \"Abundance\", \"id\": \"3dc35c18d08ff25a412e460c0528ff8dd280b8ded10a98d054a7d3a757c53a613bad98013646fccb3b50a5af76a938888026e21103341f2fe5a97e99de2d2470\", \"name\": \"aggresome\", \"namespace\": \"GO\"}, {\"bel\": \"a(GO:microtubule)\", \"function\": \"Abundance\", \"id\": \"0c753f37efe0eb149a47a1dd700eda093a03b462374275c68480b6aed3be0f81aa9793a179dfd97f8e0834f1fb9be4b4a6cff2165b176d9e8a8744f5e0d5bc2c\", \"name\": \"microtubule\", \"namespace\": \"GO\"}, {\"bel\": \"a(HBP:\\\"Biphenyl-4-yl-acrylohydroxamoc acids\\\")\", \"function\": \"Abundance\", \"id\": \"15d6f3f705e0ce0de94c56ca1a69184ccacde61260090d22b1c3b10ec291f86105257513dfe7a776ad7a20da70a2935ee5dbb9cffa25c074751bfc6a516cab55\", \"name\": \"Biphenyl-4-yl-acrylohydroxamoc acids\", \"namespace\": \"HBP\"}, {\"bel\": \"a(HBP:\\\"Tau aggregates\\\")\", \"function\": \"Abundance\", \"id\": \"2f93b798e33773ea9681389870bbee41f6fa7ee04c087bda4bd1492520385f58d089a7261b1353e15c31d3e29a96ff31884886e6ab9fa477ef8f63f19cf154f7\", \"name\": \"Tau aggregates\", \"namespace\": \"HBP\"}, {\"bel\": \"a(HBP:HBP00093)\", \"function\": \"Abundance\", \"id\": \"40583f1741facba41ac9f3de18ebe0de8c4c07f3a19bdf18429bffb135fd6a3207e9a3b64c32f9f5547bcd6ac00f8697383aa1b78b9abe1493fb7bc23d5dd87b\", \"name\": \"HBP00093\", \"namespace\": \"HBP\"}, {\"bel\": \"a(HBP:ST7612AA1)\", \"function\": \"Abundance\", \"id\": \"715f9ec0057b157aa19f4f21ff660a05bbd4055d38f08e297876a9e8d040cf46f5f27c09659a244684f502530a29224e978baf580d179b020c5d413917a2ef9d\", \"name\": \"ST7612AA1\", \"namespace\": \"HBP\"}, {\"bel\": \"a(HBP:Trithiocarbonates)\", \"function\": \"Abundance\", \"id\": \"cab44d265718d28b0fb47a5ca7c982a1fa950268c33eec66b82edb8d1a5bc66dcd04b816000b3c482edddf34aac8c955d2fafdd36edc261addadb80bd4948f95\", \"name\": \"Trithiocarbonates\", \"namespace\": \"HBP\"}, {\"bel\": \"a(PUBCHEM:132157820)\", \"function\": \"Abundance\", \"id\": \"cc302e3f10a3922d551a1e752ea920737b42e77d4252f8dd7a43403e89e07a175a4c24cd930784f9acb01492019c44d08dc925fce5420b325fdde56e321862ca\", \"name\": \"132157820\", \"namespace\": \"PUBCHEM\"}, {\"bel\": \"a(PUBCHEM:136299192)\", \"function\": \"Abundance\", \"id\": \"e6fcea9f6f16e053ad573c3d14f78c4a6bec2602a864eaec8d13caa4ec70286de1456b6156e997c81d34abfccf027ad74b4ca6256cce859fe176221051cae5c1\", \"name\": \"136299192\", \"namespace\": \"PUBCHEM\"}, {\"bel\": \"a(PUBCHEM:49850262)\", \"function\": \"Abundance\", \"id\": \"feaf560b29c26a39f1c44f6a35451f6d8e76c59f1547ee535893f2e23cd82cb9c5d806df153d05d6a058175e178ea111f6947564325140367edb1c9a954f935b\", \"name\": \"49850262\", \"namespace\": \"PUBCHEM\"}, {\"bel\": \"a(PUBCHEM:53340666)\", \"function\": \"Abundance\", \"id\": \"0ad0237ec79c717fcacceb700329a45018416a7dde687a608d7d4b71bf9bf04c187c1e2994f7554298ee37192aba3ee41648a227602b32444159111775e8691e\", \"name\": \"53340666\", \"namespace\": \"PUBCHEM\"}, {\"bel\": \"a(PUBCHEM:56650312)\", \"function\": \"Abundance\", \"id\": \"d2f24d6e66cf8ad837a201adb6983edf0a506a58d816c9a63f2fabcac3389cb5c7e790a8af0c84dc4eccb9e9bbd31575277202f937f75c2530f3341a1f70ab27\", \"name\": \"56650312\", \"namespace\": \"PUBCHEM\"}, {\"bel\": \"a(PUBCHEM:9804992)\", \"function\": \"Abundance\", \"id\": \"6f79694353be6aa83ba7a32f5210fa94c7e4e978d62c3ca8573f226930dcc00b0c32b9239953be791ba427864f8a244ef5c838bf626652f6f0ce9e00cfd322be\", \"name\": \"9804992\", \"namespace\": \"PUBCHEM\"}, {\"bel\": \"bp(GO:\\\"Hsp90 deacetylation\\\")\", \"function\": \"BiologicalProcess\", \"id\": \"88fb21dde4e97a930bf4cc8657feaf38cf6c96e84a4cbd5c531ca2c04cfac8fd700f2bff9e4ab36a738407bdf28d51cf55bef63232daaf302a39cd51fe73fb67\", \"name\": \"Hsp90 deacetylation\", \"namespace\": \"GO\"}, {\"bel\": \"bp(GO:\\\"autophagosome-lysosome fusion\\\")\", \"function\": \"BiologicalProcess\", \"id\": \"51bba14d70519100ae7a4d4bc2690a2018c2c11271cac224f10d5380ae6f09e2022b1bfb58ec29c11141d7636596f98ec44e48b1a717042bf6312e2306df8d7c\", \"name\": \"autophagosome-lysosome fusion\", \"namespace\": \"GO\"}, {\"bel\": \"bp(GO:\\\"microglial cell activation\\\")\", \"function\": \"BiologicalProcess\", \"id\": \"9e9682b78aa06bbcb9dae35006f6da468e11928c000097001cd665b1bedf3c75ef887ecd7781ef3c545cafb3780cb7afabec9bfc70e14a48149d8adee8502981\", \"name\": \"microglial cell activation\", \"namespace\": \"GO\"}, {\"bel\": \"bp(GO:\\\"proteasome-mediated ubiquitin-dependent protein catabolic process\\\")\", \"function\": \"BiologicalProcess\", \"id\": \"e31884054bf49e8c01cd68d025871de0290da5ccc81a412fdbd2a8926cf48c1aeaeeb16165f6d77ef1a7f647fdb4623057028cc40cadb2e5e4eeb5c45e00ae56\", \"name\": \"proteasome-mediated ubiquitin-dependent protein catabolic process\", \"namespace\": \"GO\"}, {\"bel\": \"bp(GO:\\\"protein K63-linked ubiquitination\\\")\", \"function\": \"BiologicalProcess\", \"id\": \"d5779ec24dd6bc44c690d1840f41d3ee7d7d45b1d5d048784a224c7a233b7c5df0be70caa0f102f77281864087bb20bba37dec639c79cff8c0f45deabbc52983\", \"name\": \"protein K63-linked ubiquitination\", \"namespace\": \"GO\"}, {\"bel\": \"bp(GO:\\\"response to stress\\\")\", \"function\": \"BiologicalProcess\", \"id\": \"6fa75459766cad2fc6b7a639d1866b76e32fe67fc60da0c4813032337226b441dfdb2ef161f19ab85095fc3aa6a4538d309d2c857d6195f021bbdf91fd6ee683\", \"name\": \"response to stress\", \"namespace\": \"GO\"}, {\"bel\": \"bp(GO:\\\"stress granule assembly\\\")\", \"function\": \"BiologicalProcess\", \"id\": \"6ff9e6cdcaae7383f1e6e6244e83c4ef2a65d31f8bd897a1702f45d6a823db11d03ac6f118557af3a72dbb2ca95073b98de81dfb2437ffcc5480f9bed181772e\", \"name\": \"stress granule assembly\", \"namespace\": \"GO\"}, {\"bel\": \"bp(GO:\\\"ubiquitin-dependent protein catabolic process\\\")\", \"function\": \"BiologicalProcess\", \"id\": \"3ed4aeb4d82f7dcc8a03e49d3bef5067d73f7e02384fda77512dde746821e34d49741cbb15b6662bac9112b47bf511dd62ab69e3834566f1c987e44bc8f421bc\", \"name\": \"ubiquitin-dependent protein catabolic process\", \"namespace\": \"GO\"}, {\"bel\": \"bp(GO:autophagy)\", \"function\": \"BiologicalProcess\", \"id\": \"f1a3202be460e565f65c1445ee061e99f63cc05f9ab70c5d4d8d93ba180d6a94f65cb8d1ac6c5ba53c0a4d2e153db32135e6a0bffcfdfd4d84e6727594a70b0a\", \"name\": \"autophagy\", \"namespace\": \"GO\"}, {\"bel\": \"complex(GO:\\\"chaperone complex\\\")\", \"function\": \"Complex\", \"id\": \"701c6898a1de0354ca25d5ec749c099a602472eecb2cf0c2bb509ff34f30149ecf5f2b0365e88b36ffcde7aaff8f64d54d3537104b67eb71351f3b168d483f92\", \"name\": \"chaperone complex\", \"namespace\": \"GO\"}, {\"bel\": \"complex(p(HGNC:G3BP1), p(HGNC:HDAC6))\", \"function\": \"Complex\", \"id\": \"ef30a13cfb55d57daa1dde012a8407621ff8ea25609149162ca046bbc983f0d9f2ec553fb0f6e7585e24bf844695f39c330f0ab768299f3c303dc406ce485dc2\", \"members\": [{\"bel\": \"p(HGNC:G3BP1)\", \"function\": \"Protein\", \"id\": \"535e7436aa458a3d7631bf1dbc0efb16251845896d493da799a1abdbe0906efb4888ed61f15d8f05bc3f8ef3af0eddabb2f3d3b32c9a4cfe3568ecf72dc9e3f0\", \"name\": \"G3BP1\", \"namespace\": \"HGNC\"}, {\"bel\": \"p(HGNC:HDAC6)\", \"function\": \"Protein\", \"id\": \"c6e2cb3c91ba614ec9e19574656eba9d234344b755f57479d99a4d4ff77acc4917594446b9b19f9823e8a8a0d2d457d74b6f6a582d9f360f13df4136d902c7c3\", \"name\": \"HDAC6\", \"namespace\": \"HGNC\"}]}, {\"bel\": \"complex(p(HGNC:HDAC6), p(HGNC:SQSTM1))\", \"function\": \"Complex\", \"id\": \"55713b4f5179418b3093225a1cf56bc0182ec9bc1edfdcd009ff2ae8d34056eddd4d10a00a389107ba45e19709a550d271b63beffbe2366c6cede2f55c9fd69a\", \"members\": [{\"bel\": \"p(HGNC:HDAC6)\", \"function\": \"Protein\", \"id\": \"c6e2cb3c91ba614ec9e19574656eba9d234344b755f57479d99a4d4ff77acc4917594446b9b19f9823e8a8a0d2d457d74b6f6a582d9f360f13df4136d902c7c3\", \"name\": \"HDAC6\", \"namespace\": \"HGNC\"}, {\"bel\": \"p(HGNC:SQSTM1)\", \"function\": \"Protein\", \"id\": \"8cd06d93cab875a139eae92082f11af5ad651146fabba6fa298a46f2cfda5bcb2bccf9dec8cc73067f5c11595d8eeb288f5c13f6a56dd316b54d108e5e66217d\", \"name\": \"SQSTM1\", \"namespace\": \"HGNC\"}]}, {\"bel\": \"complex(p(HGNC:HDAC6), p(INTERPRO:\\\"Heat shock protein Hsp90 family\\\"))\", \"function\": \"Complex\", \"id\": \"5fc301f3bdaec43d6492097421ff39454474f4f47fef7f855bb2b43088a3ae8fecf9d20c7faeb2cc696fdc8dfed60a43f044b51679f1e19c196c2e82b4bd235f\", \"members\": [{\"bel\": \"p(HGNC:HDAC6)\", \"function\": \"Protein\", \"id\": \"c6e2cb3c91ba614ec9e19574656eba9d234344b755f57479d99a4d4ff77acc4917594446b9b19f9823e8a8a0d2d457d74b6f6a582d9f360f13df4136d902c7c3\", \"name\": \"HDAC6\", \"namespace\": \"HGNC\"}, {\"bel\": \"p(INTERPRO:\\\"Heat shock protein Hsp90 family\\\")\", \"function\": \"Protein\", \"id\": \"2988f2a3c29b9ea53ca0dbbbe269b88585850e69e2c356cf8809a96231297fce42a977dbdb23678f7a3f8be0be9b59b9a38d7f416a831b8298b10fd5444ddea8\", \"name\": \"Heat shock protein Hsp90 family\", \"namespace\": \"INTERPRO\"}]}, {\"bel\": \"p(FPLX:CREB)\", \"function\": \"Protein\", \"id\": \"fe50c33cb42b7035c8602bc3ced677d7630fd479190cda030432dd4078ebe47da59de11b7b78ab3f4d0888b5839187872b3ad589a9109572fa628b6eac901edc\", \"name\": \"CREB\", \"namespace\": \"FPLX\"}, {\"bel\": \"p(FPLX:HDAC)\", \"function\": \"Protein\", \"id\": \"d2e8833ff804c3b8c5a045ac8a4096c1c66c61101c9ed5ce1f0c355bf74149b9cec6584bc3e5186e942bbd700bdafa9ccf9bab0127fad62731702159604e5ab5\", \"name\": \"HDAC\", \"namespace\": \"FPLX\"}, {\"bel\": \"p(HGNC:G3BP1)\", \"function\": \"Protein\", \"id\": \"535e7436aa458a3d7631bf1dbc0efb16251845896d493da799a1abdbe0906efb4888ed61f15d8f05bc3f8ef3af0eddabb2f3d3b32c9a4cfe3568ecf72dc9e3f0\", \"name\": \"G3BP1\", \"namespace\": \"HGNC\"}, {\"bel\": \"p(HGNC:HDAC6)\", \"function\": \"Protein\", \"id\": \"c6e2cb3c91ba614ec9e19574656eba9d234344b755f57479d99a4d4ff77acc4917594446b9b19f9823e8a8a0d2d457d74b6f6a582d9f360f13df4136d902c7c3\", \"name\": \"HDAC6\", \"namespace\": \"HGNC\"}, {\"bel\": \"p(HGNC:SQSTM1)\", \"function\": \"Protein\", \"id\": \"8cd06d93cab875a139eae92082f11af5ad651146fabba6fa298a46f2cfda5bcb2bccf9dec8cc73067f5c11595d8eeb288f5c13f6a56dd316b54d108e5e66217d\", \"name\": \"SQSTM1\", \"namespace\": \"HGNC\"}, {\"bel\": \"p(INTERPRO:\\\"Heat shock protein Hsp90 family\\\")\", \"function\": \"Protein\", \"id\": \"2988f2a3c29b9ea53ca0dbbbe269b88585850e69e2c356cf8809a96231297fce42a977dbdb23678f7a3f8be0be9b59b9a38d7f416a831b8298b10fd5444ddea8\", \"name\": \"Heat shock protein Hsp90 family\", \"namespace\": \"INTERPRO\"}, {\"bel\": \"p(MESH:Proteins)\", \"function\": \"Protein\", \"id\": \"20fec9748aca569e46ee2703620f548aece50dde1dd6518d2833a58da6a5fc28e70918b127d515943ad659b626fa533df7854b4887e2cab8d3a7af00dea602d0\", \"name\": \"Proteins\", \"namespace\": \"MESH\"}, {\"bel\": \"p(MESH:Proteins, pmod(Ub))\", \"function\": \"Protein\", \"id\": \"eec7827eaf9a8c63740d8caddb124f2f9b407f08cff90cff9b204cb3886d1f03dbc8af83ba8a0c83a87b0e7d0803537125ed0137e951a2d38e656efeac1ba0f2\", \"name\": \"Proteins\", \"namespace\": \"MESH\", \"variants\": [{\"identifier\": {\"name\": \"Ub\", \"namespace\": \"bel\"}, \"kind\": \"pmod\"}]}, {\"bel\": \"p(hgnc:AIP)\", \"function\": \"Protein\", \"id\": \"4cc9ac5dc8cf5496d3dd51725249164bc9bfc7ccecac3710bb85741227ce156b91bc628bb00421bd9d9fcd659abf0b1d59ac83da6f8f325070769e1e77edfdf3\", \"identifier\": \"358\", \"name\": \"AIP\", \"namespace\": \"hgnc\"}, {\"bel\": \"p(hgnc:CTNNB1)\", \"function\": \"Protein\", \"id\": \"f802757ae2f27bd8beff9484587217c31f6f7933eeb36e5fd1987684b0133d117ed5a954671e6bc75e575fd1faed93b78e81f2d626f686c670d61441b6ec2aaa\", \"identifier\": \"2514\", \"name\": \"CTNNB1\", \"namespace\": \"hgnc\"}, {\"bel\": \"p(hgnc:CTNNB1, pmod(Ac, Lys, 49))\", \"function\": \"Protein\", \"id\": \"6321bf108b07d97659fb41579c46a9b00a290fa4e9073c8888357de1acbcb7500e19c2a9c93d87000a326adf15bdeadcac3fdae3ce9a5c7e351989ecfb4eaff9\", \"identifier\": \"2514\", \"name\": \"CTNNB1\", \"namespace\": \"hgnc\", \"variants\": [{\"code\": \"Lys\", \"identifier\": {\"name\": \"Ac\", \"namespace\": \"bel\"}, \"kind\": \"pmod\", \"pos\": 49}]}, {\"bel\": \"p(hgnc:CTNNB1, pmod(Ph, Ser, 33))\", \"function\": \"Protein\", \"id\": \"3a4e20ccfb917141118fec49c03a5f4ea6b3211785db82f4df26a68f92d86005b313e7d2c6d0c5ac8af7b60f9aae1941939c8ec528b57d829d9ff82c4a14be4f\", \"identifier\": \"2514\", \"name\": \"CTNNB1\", \"namespace\": \"hgnc\", \"variants\": [{\"code\": \"Ser\", \"identifier\": {\"name\": \"Ph\", \"namespace\": \"bel\"}, \"kind\": \"pmod\", \"pos\": 33}]}, {\"bel\": \"p(hgnc:CTNNB1, pmod(Ph, Ser, 37))\", \"function\": \"Protein\", \"id\": \"c1b94a44c90aad545cd9548d2b37654e5c4035de95382af91e8747a091b55ebacc687242e45e3cc0f9035f2179b4d04e0b753a0f277e20b7f348afe74bd7973e\", \"identifier\": \"2514\", \"name\": \"CTNNB1\", \"namespace\": \"hgnc\", \"variants\": [{\"code\": \"Ser\", \"identifier\": {\"name\": \"Ph\", \"namespace\": \"bel\"}, \"kind\": \"pmod\", \"pos\": 37}]}, {\"bel\": \"p(hgnc:CTNNB1, pmod(Ph, Ser, 45))\", \"function\": \"Protein\", \"id\": \"fb93898d9dea20e84afcafb75f30428cd4280b6e71a9debd5948e3c97d95419e3de60a53e8bac8b9d457d40ae08165c829671b53a19aca8314d2f8c816ee921d\", \"identifier\": \"2514\", \"name\": \"CTNNB1\", \"namespace\": \"hgnc\", \"variants\": [{\"code\": \"Ser\", \"identifier\": {\"name\": \"Ph\", \"namespace\": \"bel\"}, \"kind\": \"pmod\", \"pos\": 45}]}, {\"bel\": \"p(hgnc:CTNNB1, pmod(Ph, Thr, 41))\", \"function\": \"Protein\", \"id\": \"4ab35818192f02c10582151e1c548e84ac4e50f30fb75cfe0cb643c45ee701b8cd5c44fb6173b4ee84ecc329a53fabb8425ced3efdc01852a6ed13a2aa4a4e11\", \"identifier\": \"2514\", \"name\": \"CTNNB1\", \"namespace\": \"hgnc\", \"variants\": [{\"code\": \"Thr\", \"identifier\": {\"name\": \"Ph\", \"namespace\": \"bel\"}, \"kind\": \"pmod\", \"pos\": 41}]}, {\"bel\": \"p(hgnc:HDAC3)\", \"function\": \"Protein\", \"id\": \"a0e91bf4ca13f7a6f01221b2d7d2798497a6700353caf716077d07fe95c233f4b0575d4743178cf7c8b0bc1b983bfae0e99e1cb4c78984040915df0b1fcfdfb4\", \"identifier\": \"4854\", \"name\": \"HDAC3\", \"namespace\": \"hgnc\"}, {\"bel\": \"p(hgnc:HDAC6)\", \"function\": \"Protein\", \"id\": \"9c892cb49df31546a9b99187a302a601d1bb99e3ab99477f79f128f802b2684ca782bbfe681b5edc70c98eb061eaa6a8b2633534015c84dcdc713a4bfd4a331e\", \"identifier\": \"14064\", \"name\": \"HDAC6\", \"namespace\": \"hgnc\"}, {\"bel\": \"p(hgnc:MAPT)\", \"function\": \"Protein\", \"id\": \"75ac052d10d9c36a24437908ebde27a24642cd6780dd93b2a8f6754e2eb0f4ed426269793e92438bff32aa2d71f00857c38f3db9d3d5f0ea5e108baf465920f9\", \"identifier\": \"6893\", \"name\": \"MAPT\", \"namespace\": \"hgnc\"}, {\"bel\": \"p(hgnc:MAPT, pmod(Ac))\", \"function\": \"Protein\", \"id\": \"086007502eb8870b9deadf44edabe3108b427c39b4f23cb2a48365a82798681e35859fa246d83af2d04e0514c5fe75d4b7ef04a9db976c53c41467f0c54894b5\", \"identifier\": \"6893\", \"name\": \"MAPT\", \"namespace\": \"hgnc\", \"variants\": [{\"identifier\": {\"name\": \"Ac\", \"namespace\": \"bel\"}, \"kind\": \"pmod\"}]}, {\"bel\": \"p(hgnc:MAPT, pmod(Ac, Lys))\", \"function\": \"Protein\", \"id\": \"db3dc39b4e2987d2fc6b57b379f33981aa4b39205f8aec2f7aae45ded915a3910f7f6a7764193ccd1ca221b164bcfcaea8a17fbe9b6445e342d1b8076fc3970d\", \"identifier\": \"6893\", \"name\": \"MAPT\", \"namespace\": \"hgnc\", \"variants\": [{\"code\": \"Lys\", \"identifier\": {\"name\": \"Ac\", \"namespace\": \"bel\"}, \"kind\": \"pmod\"}]}, {\"bel\": \"p(hgnc:MAPT, pmod(Ac, Lys, 321))\", \"function\": \"Protein\", \"id\": \"fadf17cfc492ba8813ad7a14a98bbf2e281c63db4c284c11be96ad37aea955f7226b33a31e947c4d1f24f72667ec237fe6a09d380e43c9245088ef7765088ebb\", \"identifier\": \"6893\", \"name\": \"MAPT\", \"namespace\": \"hgnc\", \"variants\": [{\"code\": \"Lys\", \"identifier\": {\"name\": \"Ac\", \"namespace\": \"bel\"}, \"kind\": \"pmod\", \"pos\": 321}]}, {\"bel\": \"p(hgnc:MAPT, pmod(Ph, Ser, 262))\", \"function\": \"Protein\", \"id\": \"b951b2d1026ecceeb7e3712b3fff29350f55f6ebf605b4913c8b876f604d2025542c84c59016f3d6e22d3e82400f0d6c42488351edfb0510eaeb1b406bdf3dc6\", \"identifier\": \"6893\", \"name\": \"MAPT\", \"namespace\": \"hgnc\", \"variants\": [{\"code\": \"Ser\", \"identifier\": {\"name\": \"Ph\", \"namespace\": \"bel\"}, \"kind\": \"pmod\", \"pos\": 262}]}, {\"bel\": \"p(hgnc:MAPT, pmod(Ph, Ser, 324))\", \"function\": \"Protein\", \"id\": \"7510b22fc6967ea05bd541e1945300a7efb52eb957bbd96ee3da235171d86bb7f397153a2719549b37a1e343bc46abfee872cb9908d43dc812471b3a242ed502\", \"identifier\": \"6893\", \"name\": \"MAPT\", \"namespace\": \"hgnc\", \"variants\": [{\"code\": \"Ser\", \"identifier\": {\"name\": \"Ph\", \"namespace\": \"bel\"}, \"kind\": \"pmod\", \"pos\": 324}]}, {\"bel\": \"p(hgnc:MAPT, pmod(Ph, Ser, 356))\", \"function\": \"Protein\", \"id\": \"58ce747f4a0d600d79bf5ada0ce6ca90a52d098d6894c1a73462f151fdbe74a57aaa23e2220fbbbf9032b64cdc6b38e3ad501ccc777c8e33ed0da67165ba4396\", \"identifier\": \"6893\", \"name\": \"MAPT\", \"namespace\": \"hgnc\", \"variants\": [{\"code\": \"Ser\", \"identifier\": {\"name\": \"Ph\", \"namespace\": \"bel\"}, \"kind\": \"pmod\", \"pos\": 356}]}]}, \"#uzkoxrqtjpghsaec\", 1000, 650, {'Protein': '#1F77B4', 'Pathology': '#FF7F0E', 'BiologicalProcess': '#2CA02C', 'miRNA': '#D62728', 'Complex': '#98DF8A', 'Composite': '#9467BD', 'Reaction': '#000000', 'Gene': '#FFBB78', 'Abundance': '#AEC7E8', 'RNA': '#FF9896'});\n",
       "});"
      ],
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pybel_jupyter.to_jupyter(subgraph)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "there are 15 PubMed references\n"
     ]
    }
   ],
   "source": [
    "pmid_to_targets = defaultdict(set)\n",
    "\n",
    "for u, v, data in subgraph.edges(data=True):\n",
    "    if 'citation' in data:\n",
    "        for x in u, v:\n",
    "            if 'name' in x and x['namespace'] == 'hgnc':\n",
    "                pmid_to_targets[data['citation']['reference']].add(x)\n",
    "\n",
    "pmid_to_targets = dict(pmid_to_targets)\n",
    "\n",
    "\n",
    "with open('hdac6-pmids.txt', 'w') as file:\n",
    "    for pmid in sorted(pmid_to_targets):\n",
    "        print(f'pmid:{pmid}', file=file)\n",
    "\n",
    "print(f'there are {len(pmid_to_targets)} PubMed references')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'21427723': {<BEL p(hgnc:MAPT, pmod(Ac))>},\n",
       " '18930136': {<BEL p(hgnc:HDAC6)>},\n",
       " '28803412': {<BEL p(hgnc:HDAC6)>},\n",
       " '23487739': {<BEL p(hgnc:HDAC6)>},\n",
       " '27181519': {<BEL p(hgnc:HDAC6)>},\n",
       " '26631930': {<BEL p(hgnc:HDAC6)>, <BEL p(hgnc:MAPT, pmod(Ac, Lys))>},\n",
       " '28760828': {<BEL p(hgnc:HDAC6)>,\n",
       "  <BEL p(hgnc:MAPT)>,\n",
       "  <BEL p(hgnc:MAPT, pmod(Ac, Lys, 321))>,\n",
       "  <BEL p(hgnc:MAPT, pmod(Ph, Ser, 324))>},\n",
       " '29311797': {<BEL p(hgnc:AIP)>, <BEL p(hgnc:HDAC6)>},\n",
       " '25546293': {<BEL p(hgnc:CTNNB1, pmod(Ac, Lys, 49))>,\n",
       "  <BEL p(hgnc:CTNNB1, pmod(Ph, Ser, 33))>,\n",
       "  <BEL p(hgnc:CTNNB1, pmod(Ph, Ser, 37))>,\n",
       "  <BEL p(hgnc:CTNNB1, pmod(Ph, Ser, 45))>,\n",
       "  <BEL p(hgnc:CTNNB1, pmod(Ph, Thr, 41))>,\n",
       "  <BEL p(hgnc:HDAC6)>},\n",
       " '25031639': {<BEL p(hgnc:HDAC6)>,\n",
       "  <BEL p(hgnc:MAPT, pmod(Ph, Ser, 262))>,\n",
       "  <BEL p(hgnc:MAPT, pmod(Ph, Ser, 356))>},\n",
       " '24844691': {<BEL p(hgnc:HDAC6)>},\n",
       " '18558669': {<BEL p(hgnc:HDAC6)>},\n",
       " '29844403': {<BEL p(hgnc:HDAC6)>},\n",
       " '26751493': {<BEL p(hgnc:HDAC6)>, <BEL p(hgnc:MAPT)>},\n",
       " '28771976': {<BEL p(hgnc:HDAC3)>}}"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pmid_to_targets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "pubmed_response = get_pubmed_citation_response(pmid_to_targets)\n",
    "\n",
    "pmid_pmc_map = {}\n",
    "\n",
    "for pmid in pubmed_response['result']['uids']:\n",
    "    for article_id in pubmed_response['result'][pmid]['articleids']:\n",
    "        if 'pmc' == article_id['idtype']:\n",
    "            pmid_pmc_map[pmid] = article_id['value']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hgnc.symbol</th>\n",
       "      <th>hgnc</th>\n",
       "      <th>pmid</th>\n",
       "      <th>pmc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AIP</td>\n",
       "      <td>358</td>\n",
       "      <td>29311797</td>\n",
       "      <td>PMC5744016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CTNNB1</td>\n",
       "      <td>2514</td>\n",
       "      <td>25546293</td>\n",
       "      <td>PMC4372110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HDAC3</td>\n",
       "      <td>4854</td>\n",
       "      <td>28771976</td>\n",
       "      <td>PMC5595690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>18558669</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>18930136</td>\n",
       "      <td>PMC2621359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>23487739</td>\n",
       "      <td>PMC3606981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>24844691</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>25031639</td>\n",
       "      <td>PMC4075151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>25546293</td>\n",
       "      <td>PMC4372110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>26631930</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>26751493</td>\n",
       "      <td>PMC4808800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>27181519</td>\n",
       "      <td>PMC5055418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>28760828</td>\n",
       "      <td>PMC5602388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>28803412</td>\n",
       "      <td>PMC5663814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>29311797</td>\n",
       "      <td>PMC5744016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>HDAC6</td>\n",
       "      <td>14064</td>\n",
       "      <td>29844403</td>\n",
       "      <td>PMC5974403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MAPT</td>\n",
       "      <td>6893</td>\n",
       "      <td>21427723</td>\n",
       "      <td>PMC3120096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MAPT</td>\n",
       "      <td>6893</td>\n",
       "      <td>25031639</td>\n",
       "      <td>PMC4075151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAPT</td>\n",
       "      <td>6893</td>\n",
       "      <td>26631930</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>MAPT</td>\n",
       "      <td>6893</td>\n",
       "      <td>26751493</td>\n",
       "      <td>PMC4808800</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MAPT</td>\n",
       "      <td>6893</td>\n",
       "      <td>28760828</td>\n",
       "      <td>PMC5602388</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   hgnc.symbol   hgnc      pmid         pmc\n",
       "8          AIP    358  29311797  PMC5744016\n",
       "12      CTNNB1   2514  25546293  PMC4372110\n",
       "5        HDAC3   4854  28771976  PMC5595690\n",
       "2        HDAC6  14064  18558669        None\n",
       "16       HDAC6  14064  18930136  PMC2621359\n",
       "14       HDAC6  14064  23487739  PMC3606981\n",
       "4        HDAC6  14064  24844691        None\n",
       "15       HDAC6  14064  25031639  PMC4075151\n",
       "13       HDAC6  14064  25546293  PMC4372110\n",
       "19       HDAC6  14064  26631930        None\n",
       "3        HDAC6  14064  26751493  PMC4808800\n",
       "17       HDAC6  14064  27181519  PMC5055418\n",
       "0        HDAC6  14064  28760828  PMC5602388\n",
       "10       HDAC6  14064  28803412  PMC5663814\n",
       "9        HDAC6  14064  29311797  PMC5744016\n",
       "18       HDAC6  14064  29844403  PMC5974403\n",
       "7         MAPT   6893  21427723  PMC3120096\n",
       "6         MAPT   6893  25031639  PMC4075151\n",
       "1         MAPT   6893  26631930        None\n",
       "20        MAPT   6893  26751493  PMC4808800\n",
       "11        MAPT   6893  28760828  PMC5602388"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(\n",
    "    data=list({\n",
    "        (\n",
    "            target.name,\n",
    "            target.identifier,\n",
    "            pmid,\n",
    "            pmid_pmc_map.get(pmid),\n",
    "        )\n",
    "    for pmid, targets in pmid_to_targets.items()\n",
    "    for target in targets\n",
    "    }), \n",
    "    columns=['hgnc.symbol', 'hgnc', 'pmid', 'pmc']\n",
    ").sort_values(['hgnc.symbol', 'pmid'])\n",
    "\n",
    "# Output\n",
    "df.to_csv('hdac6-article-targets.tsv', sep='\\t', index=False)\n",
    "\n",
    "# Show in notebook\n",
    "df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
