{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import time\n",
    "\n",
    "from pybel.struct import (\n",
    "    get_subgraph_by_induction, expand_upstream_causal, get_upstream_causal_subgraph, has_protein_modification,\n",
    ")\n",
    "import hbp_knowledge\n",
    "from pybel.dsl import Protein\n",
    "from pybel_jupyter import to_jupyter\n",
    "from pybel.constants import RELATION, HAS_VARIANT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'0.0.2'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hbp_knowledge.VERSION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3.7.1 (default, Nov 28 2018, 11:51:47) \n",
      "[Clang 10.0.0 (clang-1000.11.45.5)]\n"
     ]
    }
   ],
   "source": [
    "print(sys.version)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sun Jan 13 19:10:20 2019\n"
     ]
    }
   ],
   "source": [
    "print(time.asctime())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "albuquerque2009 v1.0.0\n",
      "Number of Nodes: 1934\n",
      "Number of Edges: 5437\n",
      "Network Density: 1.45E-03\n",
      "Number of Components: 18\n",
      "Number of Warnings: 0\n"
     ]
    }
   ],
   "source": [
    "graph = hbp_knowledge.get_graph()\n",
    "graph.summarize()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "List all nodes that have to do with ``tau``."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "a(HBP:\"3R tau\")\n",
      "a(HBP:\"4R tau\")\n",
      "a(HBP:\"Tau aggregates\")\n",
      "a(HBP:\"Tau annular protofibrils\")\n",
      "a(HBP:\"Tau antibody, 40E8\")\n",
      "a(HBP:\"Tau antibody, 4E4\")\n",
      "a(HBP:\"Tau antibody, 6C5\")\n",
      "a(HBP:\"Tau antibody, HT7\")\n",
      "a(HBP:\"Tau antibody, TNT1\")\n",
      "a(HBP:\"Tau antibody, Tau-13\")\n",
      "a(HBP:\"Tau antibody, Tau46\")\n",
      "a(HBP:\"Tau antibody, pS396\")\n",
      "a(HBP:\"Tau dimers\")\n",
      "a(HBP:\"Tau epitope, 12E8\")\n",
      "a(HBP:\"Tau epitope, AT180\")\n",
      "a(HBP:\"Tau epitope, AT8\")\n",
      "a(HBP:\"Tau epitope, PHF1\")\n",
      "a(HBP:\"Tau fibrils\")\n",
      "a(HBP:\"Tau isoform B (381 aa)\")\n",
      "a(HBP:\"Tau isoform C (410 aa)\")\n",
      "a(HBP:\"Tau isoform D (383 aa)\")\n",
      "a(HBP:\"Tau isoform E (412 aa)\")\n",
      "a(HBP:\"Tau isoform F (441 aa)\")\n",
      "a(HBP:\"Tau isoform Fetal-tau (352 aa)\")\n",
      "a(HBP:\"Tau oligomers\")\n",
      "a(HBP:\"granular tau oligomers\")\n",
      "a(HBP:\"sonicated tau fibrils\")\n",
      "p(HBP:\"6D tau\", frag(\"2_18\"))\n",
      "p(HBP:\"6D tau\")\n",
      "p(HBP:\"Tau isoform F (441 aa)\", var(\"p.Ile277Pro\"), var(\"p.Ile308Pro\"), var(\"p.Lys280del\"))\n",
      "p(HBP:\"Tau isoform F (441 aa)\")\n",
      "p(HBP:\"Tau isoform F (441 aa)\", var(\"p.Lys280del\"))\n",
      "p(HBP:\"6P tau\")\n",
      "p(HBP:\"AT8 tau\")\n",
      "p(HBP:\"AT8 tau\", pmod(Ph))\n",
      "p(HBP:\"Tau aggregates\")\n",
      "p(HBP:\"Tau fibrils\")\n",
      "p(HBP:\"Tau isoform B (381 aa)\")\n",
      "p(HBP:\"Tau isoform B (381 aa)\", pmod(HBP:\"protein aggregation\"))\n",
      "p(HBP:\"Tau isoform C (410 aa)\")\n",
      "p(HBP:\"Tau isoform C (410 aa)\", pmod(HBP:\"protein aggregation\"))\n",
      "p(HBP:\"Tau isoform D (383 aa)\")\n",
      "p(HBP:\"Tau isoform D (383 aa)\", pmod(HBP:\"protein aggregation\"))\n",
      "p(HBP:\"Tau isoform E (412 aa)\")\n",
      "p(HBP:\"Tau isoform E (412 aa)\", pmod(HBP:\"protein aggregation\"))\n",
      "p(HBP:\"Tau isoform F (441 aa)\", pmod(HBP:\"protein aggregation\"))\n",
      "p(HBP:\"Tau isoform Fetal-tau (352 aa)\")\n",
      "p(HBP:\"Tau isoform Fetal-tau (352 aa)\", pmod(HBP:\"protein aggregation\"))\n",
      "p(HBP:\"Tau oligomers\")\n",
      "p(HBP:\"Tau oligomers\", pmod(HBP:hyperphosphorylation))\n",
      "p(HBP:\"delta 144-273 tau\")\n",
      "p(HBP:\"tau aggregates\")\n",
      "path(MESH:Tauopathies)\n",
      "path(MESHD:Tauopathies)\n"
     ]
    }
   ],
   "source": [
    "nodes = []\n",
    "for node in graph.nodes():\n",
    "    if 'name' in node and 'tau' in node.name.lower():\n",
    "        print(node)\n",
    "        nodes.append(node)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get post-translational modifications of the human Tau protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<BEL p(HGNC:MAPT, pmod(Ph), var(\"p.S422E\"))>,\n",
       " <BEL p(HGNC:MAPT, pmod(HBP:HBP00007))>,\n",
       " <BEL p(HGNC:MAPT, pmod(HBP:hyperphosphorylation))>,\n",
       " <BEL p(HGNC:MAPT, pmod(Ph))>,\n",
       " <BEL p(HGNC:MAPT, pmod(Ph, Ser, 199))>,\n",
       " <BEL p(HGNC:MAPT, pmod(Ph, Ser, 202))>,\n",
       " <BEL p(HGNC:MAPT, pmod(Ph, Ser, 396))>,\n",
       " <BEL p(HGNC:MAPT, pmod(Ph, Ser, 404))>,\n",
       " <BEL p(HGNC:MAPT, pmod(Ph, Ser, 422))>,\n",
       " <BEL p(HGNC:MAPT, pmod(Ph, Thr, 181))>,\n",
       " <BEL p(HGNC:MAPT, pmod(Ph, Thr, 205))>]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapt = Protein('HGNC', 'MAPT')\n",
    "nodes = [\n",
    "    v\n",
    "    for u, v, key, data in graph.edges(keys=True, data=True)\n",
    "    if u == mapt and data[RELATION] == HAS_VARIANT and has_protein_modification(v)\n",
    "]\n",
    "nodes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Induce a subgraph upstream of all PTMs of the human Tau protein."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MAPT Post-Translational Modification Controllers v0.0.2\n",
      "Number of Nodes: 27\n",
      "Number of Edges: 26\n",
      "Network Density: 3.70E-02\n",
      "Number of Components: 2\n",
      "Number of Warnings: 0\n"
     ]
    },
    {
     "data": {
      "application/javascript": [
       "function init_d3_force(d3, graph, chart, width, height, function_colors) {\n",
       "    var focus_node = null;\n",
       "    var highlight_node = null;\n",
       "\n",
       "    // Highlight color variables\n",
       "\n",
       "    // Highlight color of the node boundering\n",
       "    const highlight_node_boundering = \"#4EB2D4\";\n",
       "\n",
       "    // Highlight color of the edge\n",
       "    const highlighted_link_color = \"#4EB2D4\";\n",
       "\n",
       "    // Text highlight color\n",
       "    const highlight_text = \"#4EB2D4\";\n",
       "\n",
       "    // Size when zooming scale\n",
       "    var size = d3.scalePow().exponent(1)\n",
       "        .domain([1, 100])\n",
       "        .range([8, 24]);\n",
       "\n",
       "    // Simulation parameters\n",
       "    const linkDistance = 100;\n",
       "    const fCharge = -1000;\n",
       "    const linkStrength = 0.7;\n",
       "    const collideStrength = 1;\n",
       "\n",
       "    // Simulation defined with variables\n",
       "    var simulation = d3.forceSimulation()\n",
       "        .force(\"link\", d3.forceLink()\n",
       "            .distance(linkDistance)\n",
       "            .strength(linkStrength)\n",
       "        )\n",
       "        .force(\"collide\", d3.forceCollide()\n",
       "            .radius(function (d) {\n",
       "                return d.r + 10\n",
       "            })\n",
       "            .strength(collideStrength)\n",
       "        )\n",
       "        .force(\"charge\", d3.forceManyBody()\n",
       "            .strength(fCharge)\n",
       "        )\n",
       "        .force(\"center\", d3.forceCenter(width / 2, height / 2))\n",
       "        .force(\"y\", d3.forceY(0))\n",
       "        .force(\"x\", d3.forceX(0));\n",
       "\n",
       "    // Pin down functionality\n",
       "    var node_drag = d3.drag()\n",
       "        .on(\"start\", dragstarted)\n",
       "        .on(\"drag\", dragged)\n",
       "        .on(\"end\", dragended);\n",
       "\n",
       "    function dragstarted(d) {\n",
       "        if (!d3.event.active) simulation.alphaTarget(0.3).restart();\n",
       "        d.fx = d.x;\n",
       "        d.fy = d.y;\n",
       "    }\n",
       "\n",
       "    function dragged(d) {\n",
       "        d.fx = d3.event.x;\n",
       "        d.fy = d3.event.y;\n",
       "    }\n",
       "\n",
       "    function dragended(d) {\n",
       "        if (!d3.event.active) simulation.alphaTarget(0);\n",
       "    }\n",
       "\n",
       "    function releasenode(d) {\n",
       "        d.fx = null;\n",
       "        d.fy = null;\n",
       "    }\n",
       "\n",
       "    //END Pin down functionality\n",
       "\n",
       "\n",
       "    /**\n",
       "     * Gets the best name for a node object\n",
       "     * @param {object} nodeData object\n",
       "     * @returns {str} canonical name of the node\n",
       "     */\n",
       "    function getCanonicalName(nodeData) {\n",
       "        if (nodeData.cname) {\n",
       "            return nodeData.cname;\n",
       "        } else if (nodeData.name) {\n",
       "            return nodeData.name\n",
       "        } else if (nodeData.bel) {\n",
       "            return nodeData.bel\n",
       "        } else {\n",
       "            console.log('Undefined node: ' + nodeData);\n",
       "            return 'UNDEFINED'\n",
       "        }\n",
       "    }\n",
       "\n",
       "    const color_circunferencia = \"black\";\n",
       "    const default_link_color = \"#AAAAAA\";\n",
       "    const nominal_base_node_size = 8;\n",
       "\n",
       "    // Normal and highlighted stroke of the links (double the width of the link when highlighted)\n",
       "    const nominal_stroke = 1.5;\n",
       "\n",
       "    // Zoom variables\n",
       "    const min_zoom = 0.1;\n",
       "    const max_zoom = 10;\n",
       "\n",
       "    var svg = d3.select(chart).append(\"svg\")\n",
       "        .attr(\"width\", width)\n",
       "        .attr(\"height\", height);\n",
       "\n",
       "    // // Create definition for arrowhead.\n",
       "    svg.append(\"defs\").append(\"marker\")\n",
       "        .attr(\"id\", \"arrowhead\")\n",
       "        .attr(\"viewBox\", \"0 -5 10 10\")\n",
       "        .attr(\"refX\", 20)\n",
       "        .attr(\"refY\", 0)\n",
       "        .attr(\"markerUnits\", \"strokeWidth\")\n",
       "        .attr(\"markerWidth\", 6)\n",
       "        .attr(\"markerHeight\", 6)\n",
       "        .attr(\"orient\", \"auto\")\n",
       "        .append(\"path\")\n",
       "        .attr(\"d\", \"M0,-5L10,0L0,5\");\n",
       "\n",
       "    // // Create definition for stub.\n",
       "    svg.append(\"defs\").append(\"marker\")\n",
       "        .attr(\"id\", \"stub\")\n",
       "        .attr(\"viewBox\", \"-1 -5 2 10\")\n",
       "        .attr(\"refX\", 15)\n",
       "        .attr(\"refY\", 0)\n",
       "        .attr(\"markerUnits\", \"strokeWidth\")\n",
       "        .attr(\"markerWidth\", 6)\n",
       "        .attr(\"markerHeight\", 6)\n",
       "        .attr(\"orient\", \"auto\")\n",
       "        .append(\"path\")\n",
       "        .attr(\"d\", \"M 0,0 m -1,-5 L 1,-5 L 1,5 L -1,5 Z\");\n",
       "\n",
       "    // Background\n",
       "    svg.append(\"rect\")\n",
       "        .attr(\"width\", \"100%\")\n",
       "        .attr(\"height\", \"100%\")\n",
       "        .attr(\"fill\", \"#ffffff\")\n",
       "        .style(\"pointer-events\", \"all\")\n",
       "        // Zoom + panning functionality\n",
       "        .call(d3.zoom()\n",
       "            .scaleExtent([min_zoom, max_zoom])\n",
       "            .on(\"zoom\", zoomed))\n",
       "        .on(\"dblclick.zoom\", null);\n",
       "\n",
       "\n",
       "    function zoomed() {\n",
       "        g.attr(\"transform\", d3.event.transform);\n",
       "    }\n",
       "\n",
       "    // g = svg object where the graph will be appended\n",
       "    var g = svg.append(\"g\");\n",
       "\n",
       "    var linkedByIndex = {};\n",
       "    graph.links.forEach(function (d) {\n",
       "        linkedByIndex[d.source + \",\" + d.target] = true;\n",
       "    });\n",
       "\n",
       "    function isConnected(a, b) {\n",
       "        return linkedByIndex[a.index + \",\" + b.index] || linkedByIndex[b.index + \",\" + a.index] || a.index == b.index;\n",
       "    }\n",
       "\n",
       "    function ticked() {\n",
       "        link.attr(\"x1\", function (d) {\n",
       "            return d.source.x;\n",
       "        })\n",
       "            .attr(\"y1\", function (d) {\n",
       "                return d.source.y;\n",
       "            })\n",
       "            .attr(\"x2\", function (d) {\n",
       "                return d.target.x;\n",
       "            })\n",
       "            .attr(\"y2\", function (d) {\n",
       "                return d.target.y;\n",
       "            });\n",
       "\n",
       "        node\n",
       "            .attr(\"transform\", function (d) {\n",
       "                return \"translate(\" + d.x + \", \" + d.y + \")\";\n",
       "            });\n",
       "    }\n",
       "\n",
       "\n",
       "    simulation\n",
       "        .nodes(graph.nodes)\n",
       "        .on(\"tick\", ticked);\n",
       "\n",
       "    simulation.force(\"link\")\n",
       "        .links(graph.links);\n",
       "\n",
       "    // Definition of links nodes text...\n",
       "\n",
       "    var link = g.selectAll(\".link\")\n",
       "        .data(graph.links)\n",
       "        .enter().append(\"line\")\n",
       "        .style(\"stroke-width\", nominal_stroke)\n",
       "        .style(\"stroke\", default_link_color)\n",
       "        .style(\"stroke-dasharray\", function (d) {\n",
       "            if (['decreases', 'directlyDecreases', 'increases', 'directlyIncreases', 'negativeCorrelation',\n",
       "                'positiveCorrelation'].indexOf(d.relation) >= 0) {\n",
       "                return \"none\"\n",
       "            } else {\n",
       "                return \"4, 4\"\n",
       "            }\n",
       "        })\n",
       "        .attr(\"marker-start\", function (d) {\n",
       "            if ('positiveCorrelation' == d.relation) {\n",
       "                return \"url(#arrowhead)\"\n",
       "            }\n",
       "            else if ('negativeCorrelation' == d.relation) {\n",
       "                return \"url(#stub)\"\n",
       "            }\n",
       "            else {\n",
       "                return \"\"\n",
       "            }\n",
       "        })\n",
       "        .attr(\"marker-end\", function (d) {\n",
       "            if (['increases', 'directlyIncreases', 'positiveCorrelation', 'isA', 'partOf'].indexOf(d.relation) >= 0) {\n",
       "                return \"url(#arrowhead)\"\n",
       "            } else if (['decreases', 'directlyDecreases', 'negativeCorrelation'].indexOf(d.relation) >= 0) {\n",
       "                return \"url(#stub)\"\n",
       "            } else {\n",
       "                return \"\"\n",
       "            }\n",
       "        });\n",
       "\n",
       "    var node = g.selectAll(\".nodes\")\n",
       "        .data(graph.nodes)\n",
       "        .enter().append(\"g\")\n",
       "        .attr(\"class\", \"node\")\n",
       "        // Next two lines -> Pin down functionality\n",
       "        .on('dblclick', releasenode)\n",
       "        .call(node_drag);\n",
       "\n",
       "    var circle = node.append(\"path\")\n",
       "        .attr(\"d\", d3.symbol()\n",
       "            .size(function (d) {\n",
       "                return Math.PI * Math.pow(size(d.size) || nominal_base_node_size, 2);\n",
       "            })\n",
       "        )\n",
       "        .attr(\"class\", function (d) {\n",
       "            return d.function\n",
       "        })\n",
       "        .style('fill', function (d) {\n",
       "            return function_colors[d.function]\n",
       "        })\n",
       "        .style(\"stroke-width\", nominal_stroke)\n",
       "        .style(\"stroke\", color_circunferencia);\n",
       "\n",
       "    var text = node.append(\"text\")\n",
       "        .attr(\"class\", \"node-name\")\n",
       "        // .attr(\"id\", nodehashes[d])\n",
       "        .attr(\"fill\", \"black\")\n",
       "        .attr(\"dx\", 12)\n",
       "        .attr(\"dy\", \".35em\")\n",
       "        .text(function (d) {\n",
       "            return getCanonicalName(d)\n",
       "        });\n",
       "\n",
       "    // Highlight on mouseenter and back to normal on mouseout\n",
       "    node.on(\"mouseenter\", function (d) {\n",
       "        set_highlight(d);\n",
       "    })\n",
       "        .on(\"mousedown\", function () {\n",
       "            d3.event.stopPropagation();\n",
       "        }).on(\"mouseout\", function () {\n",
       "        exit_highlight();\n",
       "    });\n",
       "\n",
       "    function exit_highlight() {\n",
       "        highlight_node = null;\n",
       "        if (focus_node === null) {\n",
       "            if (highlight_node_boundering != color_circunferencia) {\n",
       "                circle.style(\"stroke\", color_circunferencia);\n",
       "                text.style(\"fill\", \"black\");\n",
       "                link.style(\"stroke\", default_link_color);\n",
       "            }\n",
       "        }\n",
       "    }\n",
       "\n",
       "    function set_highlight(d) {\n",
       "        if (focus_node !== null) d = focus_node;\n",
       "        highlight_node = d;\n",
       "\n",
       "        if (highlight_node_boundering != color_circunferencia) {\n",
       "            circle.style(\"stroke\", function (o) {\n",
       "                return isConnected(d, o) ? highlight_node_boundering : color_circunferencia;\n",
       "            });\n",
       "            text.style(\"fill\", function (o) {\n",
       "                return isConnected(d, o) ? highlight_text : \"black\";\n",
       "            });\n",
       "            link.style(\"stroke\", function (o) {\n",
       "                return o.source.index == d.index || o.target.index == d.index ? highlighted_link_color : default_link_color;\n",
       "            });\n",
       "        }\n",
       "    }\n",
       "\n",
       "\n",
       "    // Freeze the graph when space is pressed\n",
       "    function freezeGraph() {\n",
       "        // Space button Triggers STOP\n",
       "        if (d3.event.keyCode == 32) {\n",
       "            simulation.stop();\n",
       "        }\n",
       "    }\n",
       "\n",
       "    // Call freezeGraph when a key is pressed, freezeGraph checks whether this key is \"Space\" that triggers the freeze\n",
       "    d3.select(window).on(\"keydown\", freezeGraph);\n",
       "}\n",
       "\n",
       "require.config({\n",
       "    paths: {\n",
       "        d3: '//cdnjs.cloudflare.com/ajax/libs/d3/4.5.0/d3.min'\n",
       "    }\n",
       "});\n",
       "\n",
       "var elementInnerHTML = \"<div id='zvtuopsqryxfgcbe'></div>\";\n",
       "\n",
       "element.append(elementInnerHTML);\n",
       "\n",
       "require(['d3'], function (d3) {\n",
       "    return init_d3_force(d3, {\"directed\": true, \"graph\": {\"annotation_list\": {\"AssayType\": [\"B\"], \"HasQuantitativeInformation\": [\"False\", \"True\"], \"MeasurementRelation\": [\"=\", \"~\"], \"MeasurementType\": [\"IC50\", \"Ki\"], \"MeasurementUnits\": [\"nM\", \"μM\"], \"MeasurementValue\": [\"1.3\", \"1.4\", \"1.9\"], \"PublicationType\": [\"Comment\", \"Research\", \"Review\"], \"Published\": [\"CDR story\", \"Epilepsy comorbidity paper\"]}, \"annotation_pattern\": {\"ArticleCompoundNumber\": \".*\", \"Figure\": \".*\", \"MeasurementRangeLower\": \".*\", \"MeasurementRangeUpper\": \".*\", \"MeasurementValue\": \".*\", \"Page\": \".*\", \"Species\": \"^\\\\d+$\"}, \"annotation_url\": {\"Anatomy\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/anatomy/anatomy-20170511.belanno\", \"Cell\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/cell/cell-20170511.belanno\", \"CellLine\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/cell-line/cell-line-20170511.belanno\", \"CellStructure\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/cell-structure/cell-structure-20170511.belanno\", \"Confidence\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/confidence/confidence-1.0.0.belanno\", \"Disease\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/disease/disease-20170511.belanno\", \"Gender\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/gender/gender-1.0.0.belanno\", \"MeSHAnatomy\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/mesh-anatomy/mesh-anatomy-20170511.belanno\", \"MeSHDisease\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/mesh-diseases/mesh-diseases-20170511.belanno\", \"Species\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/species-taxonomy-id/species-taxonomy-id-20170511.belanno\", \"Subgraph\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/neurommsig/neurommsig-1.0.3.belanno\", \"TextLocation\": \"https://arty.scai.fraunhofer.de/artifactory/bel/annotation/text-location/text-location-1.0.1.belanno\"}, \"document_metadata\": {}, \"namespace_pattern\": {\"CL\": \"^\\\\d{7}$\", \"DBSNP\": \"rs[0-9]+\", \"DNSNP\": \"rs[0-9]+\", \"ECCODE\": \"^\\\\d+\\\\.-\\\\.-\\\\.-|\\\\d+\\\\.\\\\d+\\\\.-\\\\.-|\\\\d+\\\\.\\\\d+\\\\.\\\\d+\\\\.-|\\\\d+\\\\.\\\\d+\\\\.\\\\d+\\\\.(n)?\\\\d+$\", \"HBP\": \".*\", \"PUBCHEM\": \"^\\\\d+$\", \"TAXONOMY\": \"^\\\\d+$\", \"UNIPROT\": \".*\", \"dbSNP\": \"rs[0-9]+\"}, \"namespace_url\": {\"CHEBI\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/chebi-names.belns\", \"CTO\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/clinical-trial-ontology/clinical-trial-ontology-1.0.0.belns\", \"DO\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/disease-ontology/disease-ontology-20170725.belns\", \"DRUGBANK\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/drugbank-names.belns\", \"ECCODE\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/ec-code.belns\", \"FPLX\": \"https://raw.githubusercontent.com/sorgerlab/famplex/e8ae9926ff95266032cb74f77973c84939bffbeb/export/famplex.belns\", \"GFAM\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/hgnc.genefamily-names.belns\", \"GO\": \"https://raw.githubusercontent.com/pharmacome/terminology/73688d6dc24e309fca59a1340dc9ee971e9f3baa/external/go-names.belns\", \"GOBP\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/go-biological-process/go-biological-process-20170725.belns\", \"GOCC\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/go-cellular-component/go-cellular-component-20170725.belns\", \"HBP\": \"https://raw.githubusercontent.com/pharmacome/terminology/785431732a26d6809b4162d98b95687d16404e63/export/hbp.belns\", \"HGNC\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/hgnc-names.belns\", \"HGNCGENEFAMILY\": \"https://raw.githubusercontent.com/pharmacome/terminology/3074b85b858455d8eeb76cfcdef685ced19bbe11/external/hgnc.genefamily-names.belns\", \"HP\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/hp/hp-20171108.belns\", \"INTERPRO\": \"https://raw.githubusercontent.com/pharmacome/terminology/f2f993e599694ab5ce989cc39d789a499f75db99/external/interpro-names.belns\", \"MESH\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/mesh-names.belns\", \"MESHC\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/mesh-chemicals/mesh-chemicals-20170725.belns\", \"MESHCS\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/mesh-cell-structures/mesh-cell-structures-20170511.belns\", \"MESHD\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/mesh-diseases/mesh-diseases-20170725.belns\", \"MESHPP\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/mesh-processes/mesh-processes-20170725.belns\", \"MGI\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/mgi-mouse-genes/mgi-mouse-genes-20170725.belns\", \"MIRBASE\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/mirbase-names.belns\", \"NCBIGENE\": \"https://raw.githubusercontent.com/pharmacome/terminology/b46b65c3da259b6e86026514dfececab7c22a11b/external/entrez.belns\", \"PFAM\": \"https://raw.githubusercontent.com/pharmacome/terminology/8ccfed235e418e4c8aa576f9a5ef0f838e794c7f/external/pfam-names.belns\", \"RGD\": \"https://raw.githubusercontent.com/pharmacome/terminology/f2f993e599694ab5ce989cc39d789a499f75db99/external/rgd-names.belns\", \"SCOMP\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/selventa-named-complexes/selventa-named-complexes-20170725.belns\", \"SFAM\": \"https://arty.scai.fraunhofer.de/artifactory/bel/namespace/selventa-protein-families/selventa-protein-families-20170725.belns\"}, \"namespaces_uncached\": [], \"pybel_version\": \"0.13.0\"}, \"links\": [{\"citation\": {\"reference\": \"29626319\", \"type\": \"PubMed\"}, \"evidence\": \"Epigallocatechin-3-gallate enhances the clearance of ADrelevant phosphorylated tau species via increasing mRNA expression of autophagy adaptor proteins NDP52 and p62 (Chesser et al. 2016)\", \"key\": \"461e8d847795901e01de5b48d05495306fff261f7e154c8d4b56503cb5022913ffaf5e1c80a4a1fcbc7a3d1e6df80d3cdbf5127bf7a3923ea84017dfa5a82857\", \"line\": 684, \"relation\": \"decreases\", \"source\": 0, \"target\": 21}, {\"annotations\": {\"Confidence\": {\"Medium\": true}}, \"citation\": {\"reference\": \"21718217\", \"type\": \"PubMed\"}, \"evidence\": \" Treatment of 5-month-old APP/PS1 mice for 8 weeks with ATRA (20 mg/kg/day) resulted in significant decreases in Ab deposition and tau phosphorylation in these mice. Additionally, it attenuated memory deficits seen in the Morris water maze [77]. \", \"key\": \"d5f5618fd53b399630eafa5dc006debe445d56d59a8690f17360a5cfb3732a3645aef32d52465f2199e113255295aab77f0058f68d23522dbd9a42330eb56213\", \"line\": 324, \"relation\": \"decreases\", \"source\": 1, \"target\": 21}, {\"annotations\": {\"MeSHDisease\": {\"Alzheimer Disease\": true}}, \"citation\": {\"reference\": \"26813123\", \"type\": \"PubMed\"}, \"evidence\": \"However, further studies have demonstrated that nicotinic receptor activation can lead to an increase in Aβ-mediated tau phosphorylation\", \"key\": \"cf59c400bbac3d4faec345393d5367df28327b7633bb637cdbcb8aeb6989f2e5db4d64ebdac8ab79d42fd648d05f086bf47a57cc663991f1b0152020de491827\", \"line\": 706, \"relation\": \"regulates\", \"source\": 2, \"target\": 21}, {\"annotations\": {\"Confidence\": {\"High\": true}, \"MeSHDisease\": {\"Alzheimer Disease\": true}, \"Subgraph\": {\"MAPK-JNK subgraph\": true, \"Nitric oxide subgraph\": true, \"Tau protein subgraph\": true}}, \"citation\": {\"reference\": \"27314526\", \"type\": \"PubMed\"}, \"evidence\": \" Additionally, it can induce phosphorylation of the tau protein and promote for- mation of neuroﬁbrillary tangles through the mitogen activated protein kinases-p38 (MAPK-p38) stress pathway [22, 54].\", \"key\": \"bc536ed7b70306f2719258f84430b8bea7fcc130de630fd8449a6a898a7f528950fe40b0e718806217dfe3b95cff140ab58e31c087f2395345c64db4d4405b76\", \"line\": 221, \"relation\": \"increases\", \"source\": 3, \"target\": 21}, {\"citation\": {\"reference\": \"29626319\", \"type\": \"PubMed\"}, \"evidence\": \"Wogonin, rapamycin, and temsirolimus have been considered to improve the activity of autophagy to increase Aβ clearance and inhibit tau phosphorylation via targeting mTOR signaling (Caccamo et al. 2010; Jiang et al. 2014c; Jiang et al. 2014d; Spilman et al. 2010; Zhu andWang 2015)\", \"key\": \"ec10221f8c9f087affb2e687da739617142c86070990c9381a05db518472ac1d1c6f7db83e95190252805bb2db5d50cea72234574ba547393f3e36f1a0b8f582\", \"line\": 659, \"relation\": \"decreases\", \"source\": 4, \"target\": 21}, {\"annotations\": {\"MeSHDisease\": {\"Machado-Joseph Disease\": true}}, \"citation\": {\"reference\": \"30116051\", \"type\": \"PubMed\"}, \"evidence\": \"Cilostazol improved cognition and reduced levels of Aβ42 and hyperphosphorylated tau following intra- cerebroventricular injection of Aβ25–35 into mice 162,163 .\", \"key\": \"2529efcb2ffa7879985979f5dd2857fa8ad695c3b63e075df91204794ab7d817dc04ead54b8dd7de805e149e7e286eeb645c74cf982417bb07d7ba47209c0967\", \"line\": 669, \"relation\": \"decreases\", \"source\": 5, \"target\": 19}, {\"citation\": {\"reference\": \"30116051\", \"type\": \"PubMed\"}, \"evidence\": \"Among compounds that inhibit HSP90, geldanamycin promoted elimination of both hyper- phosphorylated tau and oligomeric α-synuclein in cell lines 219,220 . \", \"key\": \"9ee9472e11382f909bfc799ad2b280a417402ca39cc085a1fed6c02042223b7bb85e77e9dfe1b1298f7c7e3793b36094cbedc41001210c4765df551240c5bc06\", \"line\": 815, \"relation\": \"decreases\", \"source\": 6, \"target\": 19}, {\"annotations\": {\"MeSHAnatomy\": {\"Dopaminergic Neurons\": true}}, \"citation\": {\"reference\": \"30116051\", \"type\": \"PubMed\"}, \"evidence\": \" Moreover, reductions in levels of hyper phosphorylated tau and Aβ were seen in metformin- treated neurons 117,118 , while it blunted neuronal loss in a neurochemical lesion model of PD in mice 119 .\", \"key\": \"5a13e01891202bdd0b4e60d13a45c0b9f97abb0a984f9b6fb06f7c287f4f0a6814bd6bbba4dc226081965b140fc555119837148215b546f74043b2467934434c\", \"line\": 472, \"relation\": \"decreases\", \"source\": 7, \"target\": 19}, {\"citation\": {\"reference\": \"29626319\", \"type\": \"PubMed\"}, \"evidence\": \"Wogonin, rapamycin, and temsirolimus have been considered to improve the activity of autophagy to increase Aβ clearance and inhibit tau phosphorylation via targeting mTOR signaling (Caccamo et al. 2010; Jiang et al. 2014c; Jiang et al. 2014d; Spilman et al. 2010; Zhu andWang 2015)\", \"key\": \"39699cdb3d6605e81422afad24dcaeb0423e2c273ad980535aa7e91c7d5f0d600dccc6e5d5abde427344dfbf6f8f75a63ed14a545b17369fd911cd640d4b1bcc\", \"line\": 656, \"relation\": \"decreases\", \"source\": 8, \"target\": 21}, {\"citation\": {\"reference\": \"29626319\", \"type\": \"PubMed\"}, \"evidence\": \"Wogonin, rapamycin, and temsirolimus have been considered to improve the activity of autophagy to increase Aβ clearance and inhibit tau phosphorylation via targeting mTOR signaling (Caccamo et al. 2010; Jiang et al. 2014c; Jiang et al. 2014d; Spilman et al. 2010; Zhu andWang 2015)\", \"key\": \"3cda551e1439a3443cda803f0b8f065abfcf9120c4fa855663ae82b9be41256bc33fb4ede3d7ed453b2c72615cab50743dbe8503a6b826dd4e022e817ee4fea2\", \"line\": 653, \"relation\": \"decreases\", \"source\": 9, \"target\": 21}, {\"annotations\": {\"MeSHDisease\": {\"Alzheimer Disease\": true}, \"TextLocation\": {\"Results\": true}}, \"citation\": {\"reference\": \"28408124\", \"type\": \"PubMed\"}, \"evidence\": \"for example, the two phosphorylation dependent tau antibodies (40E8 and p396) were the most efficient in the human AD case with the highest level of phosphorylated tau (1266). Both 6C5 and 40E8, shown to be most effective at reducing uptake from HMW human AD brainederived tau species (Figure 3, B and C), immunostained NFTs and neuritic plaques in postmortem human AD frontal cortex sections (Figure 6); 40E8 was somewhat more reactive to neuropil threads under the conditions used.\", \"key\": \"d59e3b0e59d5b6267c7184505a384c6b17a1cadcb5bcd45f5cab5545a0fed42c1e3ec8ada74f6bac5e82d312eb0f85ce9ab1a35d84b7d3ad8f5da64b9d8d6e8c\", \"line\": 205, \"relation\": \"decreases\", \"source\": 10, \"target\": 21}, {\"annotations\": {\"MeSHDisease\": {\"Alzheimer Disease\": true}, \"TextLocation\": {\"Results\": true}}, \"citation\": {\"reference\": \"28408124\", \"type\": \"PubMed\"}, \"evidence\": \"for example, the two phosphorylation dependent tau antibodies (40E8 and p396) were the most efficient in the human AD case with the highest level of phosphorylated tau (1266). Both 6C5 and 40E8, shown to be most effective at reducing uptake from HMW human AD brainederived tau species (Figure 3, B and C), immunostained NFTs and neuritic plaques in postmortem human AD frontal cortex sections (Figure 6); 40E8 was somewhat more reactive to neuropil threads under the conditions used.\", \"key\": \"0ef5beeec898d0a5f847460daf93587894c8fcd007a53292a469cc113cfe32e9de0fb26989146093fd46682d1da103f97f2fa75bcf8ff0c5daf276391b189718\", \"line\": 206, \"relation\": \"decreases\", \"source\": 11, \"target\": 21}, {\"annotations\": {\"Species\": {\"10090\": true}}, \"citation\": {\"reference\": \"24590577\", \"type\": \"PubMed\"}, \"evidence\": \"Stimulation of M1 mAChR by two agonists, carbachol and AF102B, time- and dose-dependently decreases tau phosphorylation in PC12 cells[81]. Chronic treatment with AF267B also alleviates tau pathology in 3×Tg AD mice, possibly by activating PKC and inhibiting GSK-3beta\", \"key\": \"d3544c7b05ef35a817a6b364e1bb6bf15b10b81fa508c2f71d1efde896af628887c8e9ab53698cbac66b3ff08e15c0f884a79bd5b9665b6ce9e9236fcad315d4\", \"line\": 472, \"relation\": \"decreases\", \"source\": 12, \"target\": 21}, {\"annotations\": {\"MeSHDisease\": {\"Alzheimer Disease\": true}, \"Species\": {\"10090\": true}}, \"citation\": {\"reference\": \"24590577\", \"type\": \"PubMed\"}, \"evidence\": \"Chronic treatment with AF267B reduces Abeta plaques and tau hyperphosphorylation and rescues learning and memory impairments in 3×Tg AD mice\", \"key\": \"177ac309c0b2e64c466936d4b8ca5099ee7f4aa28d54cf0ac2a64936881cc0e56b493a1201799b17a4b80b3a54f65d7445006177c686953e88030f58b013ae95\", \"line\": 553, \"relation\": \"decreases\", \"source\": 12, \"target\": 21}, {\"annotations\": {\"MeSHDisease\": {\"Alzheimer Disease\": true}}, \"citation\": {\"reference\": \"26813123\", \"type\": \"PubMed\"}, \"evidence\": \"However, further studies have demonstrated that nicotinic receptor activation can lead to an increase in Aβ-mediated tau phosphorylation\", \"key\": \"cf9ed63847a7e2cee5766e18ac202f917c844346552ea0b906774c259d0cb5475a12a3e076633fead30790cb79fb11cd28f2646d5f40383ec2f16f030eddbe9e\", \"line\": 705, \"relation\": \"increases\", \"source\": 13, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 21}, {\"citation\": {\"reference\": \"29626319\", \"type\": \"PubMed\"}, \"evidence\": \"Neuronal PAS domain protein 4 has been found to facilitate the autophagic clearance of endogenous total and phosphorylated tau in cortical neurons of rats\", \"key\": \"88f2556cda92b2052b0db01782d798eeeb8d1b0d96befb0714473ba4d71a068c8996425909747c7f6d6b4b08a3d0b8b118f4e022176d838d98f05722ec09fd54\", \"line\": 704, \"object\": {\"modifier\": \"Degradation\"}, \"relation\": \"increases\", \"source\": 14, \"target\": 21}, {\"annotations\": {\"MeSHAnatomy\": {\"Dopaminergic Neurons\": true}, \"MeSHDisease\": {\"Machado-Joseph Disease\": true}}, \"citation\": {\"reference\": \"30116051\", \"type\": \"PubMed\"}, \"evidence\": \" Moreover, the flavonol fisetin stimulated auto- phagic degradation of phosphorylated tau in cortical neu- rons via mTORC1-dependent activation of TFEB and the cytoprotective transcription factor nuclear factor eryth- roid 2-related factor 2 (NFE2L2) 149 . Fisetin also reduced Aβ accumulation in an APP/PS1 mouse model of AD 150 . \", \"key\": \"5268e3ec3586829fb9039df46379f8180ba35cb3ddc6e4230807c8439be510b1d52cf54bef045036adc902290677c27025a5a7bdd37997015eb9d7f8e4c97dfa\", \"line\": 608, \"object\": {\"modifier\": \"Degradation\"}, \"relation\": \"increases\", \"source\": 15, \"target\": 19}, {\"annotations\": {\"MeSHDisease\": {\"Alzheimer Disease\": true}}, \"citation\": {\"reference\": \"26813123\", \"type\": \"PubMed\"}, \"evidence\": \"It has been shown that activation of M1 receptors decreases tau hyperphosphorylation via activation of PKC and inhibition of GSK-3β\", \"key\": \"21ef9f1ceb29b2806f5be3d649b498e4d4dc348917014b3313b5b43179902fb788e2f3712f58916199f2203ab74e0e27712d135d35a6cf40d871915ef1fca878\", \"line\": 671, \"relation\": \"decreases\", \"source\": 17, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 20}, {\"annotations\": {\"CellLine\": {\"PC12\": true}}, \"citation\": {\"reference\": \"24590577\", \"type\": \"PubMed\"}, \"evidence\": \"Stimulation of M1 mAChR by two agonists, carbachol and AF102B, time- and dose-dependently decreases tau phosphorylation in PC12 cells[81]. Chronic treatment with AF267B also alleviates tau pathology in 3×Tg AD mice, possibly by activating PKC and inhibiting GSK-3beta\", \"key\": \"7f713547866f40aac29443685d27505edd21bd411d618b72019bbf065a243b8f111ec46e953a4d07a9a2124cc0d0d850426ec1c9fe044d3e8495f3c888ef97da\", \"line\": 466, \"relation\": \"decreases\", \"source\": 17, \"subject\": {\"modifier\": \"Activity\"}, \"target\": 21}, {\"annotations\": {\"Confidence\": {\"High\": true}, \"MeSHDisease\": {\"Alzheimer Disease\": true}, \"Subgraph\": {\"Amyloidogenic subgraph\": true, \"Cyclin-CDK subgraph\": true, \"GSK3 subgraph\": true, \"Interleukin signaling subgraph\": true, \"Tau protein subgraph\": true}}, \"citation\": {\"reference\": \"27314526\", \"type\": \"PubMed\"}, \"evidence\": \"Pro-inﬂammatory IL-18 increases AD-associated A beta deposition in human neuron-like cells in culture [55]. IL-18 also increases the expression of glycogen synthase kinase 3 beta (GSK-3 beta ) and cyclin-dependent kinase 5, both of which are involved in hyperphos- phorylation of the tau protein [56]. \", \"key\": \"17370489d9f73bb9ce11ca7740a3c5411b2ed6119a5a6479cd594aca6301f34dbe7c7d4c536f48862aac7430774418f86475013d3d32c7ba283212c67c1d8075\", \"line\": 239, \"relation\": \"increases\", \"source\": 16, \"target\": 21}, {\"annotations\": {\"Confidence\": {\"High\": true}, \"MeSHDisease\": {\"Alzheimer Disease\": true}, \"Subgraph\": {\"Amyloidogenic subgraph\": true, \"Cyclin-CDK subgraph\": true, \"GSK3 subgraph\": true, \"Interleukin signaling subgraph\": true, \"Tau protein subgraph\": true}}, \"citation\": {\"reference\": \"27314526\", \"type\": \"PubMed\"}, \"evidence\": \"Pro-inﬂammatory IL-18 increases AD-associated A beta deposition in human neuron-like cells in culture [55]. IL-18 also increases the expression of glycogen synthase kinase 3 beta (GSK-3 beta ) and cyclin-dependent kinase 5, both of which are involved in hyperphos- phorylation of the tau protein [56]. \", \"key\": \"a38e47c8df699f73175afc3b9ab5e4588db394627de2a50163234aa0f05f601e6ac6a4549170a7051921acba4b6cbc55a7c4d31c7eff61fdf5879513309442d1\", \"line\": 238, \"relation\": \"increases\", \"source\": 18, \"target\": 21}, {\"citation\": {\"reference\": \"29626319\", \"type\": \"PubMed\"}, \"evidence\": \"Neuronal PAS domain protein 4 has been found to facilitate the autophagic clearance of endogenous total and phosphorylated tau in cortical neurons of rats\", \"key\": \"2de07e88271b8ec3a4fe889c279574fa9a4fde83baaa40ca71c54cb2a4c0a78c86bc90d87c27d2bcd421ec2ca1aa5dc77cccaf231b18129ea150d536d9b08b7b\", \"line\": 706, \"object\": {\"modifier\": \"Degradation\"}, \"relation\": \"increases\", \"source\": 22, \"target\": 21}, {\"citation\": {\"reference\": \"29626319\", \"type\": \"PubMed\"}, \"evidence\": \"For example, like Aβ, clearance of pTau/NFT also can be regulated by TFEB, which increases the activity of autophagy and lysosome (Polito et al. 2014)\", \"key\": \"3ae4e49fa952c3a7cbc6ad003d4122e0344ab9ccab764b654459098df94eb86a67ef766a1d3a6a7d62b2c262434b805101169ae78981ec4cf065cf4471f851e1\", \"line\": 589, \"object\": {\"modifier\": \"Degradation\"}, \"relation\": \"regulates\", \"source\": 23, \"target\": 21}, {\"annotations\": {\"MeSHAnatomy\": {\"Choroid Plexus\": true}}, \"citation\": {\"reference\": \"26195256\", \"type\": \"PubMed\"}, \"evidence\": \"Phosphorylation of tau by protein kinase A increases its resistance to degradation by calpain\", \"key\": \"aa79d8afc1f25faeddbf6ba759ecf8828d54752c7572e85892b767f0b8f1827be5f2985bd8baccfe63c49a0cf1ceebf8a00235f601f542063ecfafecd4152048\", \"line\": 272, \"relation\": \"increases\", \"source\": 24, \"target\": 21}, {\"annotations\": {\"MeSHAnatomy\": {\"Hippocampus\": true, \"Temporal Lobe\": true}}, \"citation\": {\"reference\": \"26195256\", \"type\": \"PubMed\"}, \"evidence\": \"Inflammation, a common feature of AD, can affect ligand affinity by making the pH more acidic, which promotes hyperphosphorylation of tau and induces conforma- tional changes in Aβ that hinder its clearance.\", \"key\": \"eafd7bc292469d95acbf40823e73fba99c38794a7cf8bc5be67e1382bf15fff24a0b6846486f2de971db69eca17ee7674e76818d0927be95ae1d1f253e5edd64\", \"line\": 224, \"relation\": \"increases\", \"source\": 25, \"target\": 21}, {\"annotations\": {\"Confidence\": {\"High\": true}, \"MeSHAnatomy\": {\"Intracellular Space\": true}, \"PublicationType\": {\"Review\": true}}, \"citation\": {\"reference\": \"22817713\", \"type\": \"PubMed\"}, \"evidence\": \"The tau protein becomes highly phosphorylated in AD and this is likely to induce a conformational change causing its detachment from microtubules and its accumulation in aggregates [3]\", \"key\": \"71f87dd440fc0eadb3b56430269a422f6dd6caba6d93b59b300e5456efb129286f7b2336748c46edd3a73d024644e678a993e060ad42fbb3f04511cdf3ba6c61\", \"line\": 101, \"relation\": \"increases\", \"source\": 26, \"target\": 21}], \"multigraph\": true, \"nodes\": [{\"function\": \"Abundance\", \"id\": \"6ab1c80d3de724e6d924795da884f193d598624c7407acbe86dd59da673081130206c19bcf6637d2372ba76d86fdbe737a1dff9f7a9b685fb82284147388c222\", \"name\": \"(-)-epigallocatechin 3-gallate\", \"namespace\": \"CHEBI\"}, {\"function\": \"Abundance\", \"id\": \"df1be1712b53ebbc3b8db7a99db20febb6ac30c8b6a3e24bb07b9b6e2f24bd7076b55afab356b4900766c8828c79a533b2abb4f9be2f6e4b2d809eabdc5f2807\", \"name\": \"all-trans-retinoic acid\", \"namespace\": \"CHEBI\"}, {\"function\": \"Abundance\", \"id\": \"a9487b29439483c6840095610dd3e496e3ca9cccc3a8d71e0385a39b570f3c0a26965ac06d83265e1c08deaae7998272d4f4c21ec0534f8be3fd8e4e57dbb3a2\", \"name\": \"amyloid-beta\", \"namespace\": \"CHEBI\"}, {\"function\": \"Abundance\", \"id\": \"fd0dcffbd4e00c161d74c931890123439e35f86a92cbcf20c1efaed0bb8b8469434bbafd8364de8822b9af6368de418ed1728b1534bb8eeb47320d5a9c895548\", \"name\": \"nitric oxide\", \"namespace\": \"CHEBI\"}, {\"function\": \"Abundance\", \"id\": \"6eb519f2ad4d46a6ed4151fc8fc609049b5a772cad25e88cf9c625d12370b056fee390606fd6d862c7544e539a500c9c5a0fc4eba471ccc4f90bb74e9f603024\", \"name\": \"Temsirolimus\", \"namespace\": \"CHEBI\"}, {\"function\": \"Abundance\", \"id\": \"9a9fc863375a0a8fb0f267c2c120528f381f2060312a574cf5a9d9c205f9530144f95335f491adec62c9337983ddce2b1f32ede9421bc887ccaf3349403116c5\", \"name\": \"cilostazol\", \"namespace\": \"CHEBI\"}, {\"function\": \"Abundance\", \"id\": \"495a46f0ccb373f38cd5fc5172590ae53a0619efd473182ef38981d4f2d7d2e6631c171ef484aac26e3891e0d5ef27e716420629b6f3c62aed15ac224ad41430\", \"name\": \"geldanamycin\", \"namespace\": \"CHEBI\"}, {\"function\": \"Abundance\", \"id\": \"8e168527ae6f3d001318f64311463a40ace1c41edb9de6f8a7763dbd9c4d9637ed7f886451804e6d45214868ad615fd57c3d6fb052bb97c661b4affe8ee2b50b\", \"name\": \"metformin\", \"namespace\": \"CHEBI\"}, {\"function\": \"Abundance\", \"id\": \"102b031a00eed78fd9e5c4236defb90a8995463980f9060493bb5fd3159a3eb666cbb2b6a33a213c749a2b279ef6d149796b4f0b568e03b82eb5c89b0da77426\", \"name\": \"sirolimus\", \"namespace\": \"CHEBI\"}, {\"function\": \"Abundance\", \"id\": \"eec47f7ee4a9d4cc98914010c6abf560803b8a6d2e0782db0d99f6f58a4b8a859c557ef3f9dccf4ec19fd4e5c71e93a5704a4331cf6c7c62fb74e27049f5b3e7\", \"name\": \"wogonin\", \"namespace\": \"CHEBI\"}, {\"function\": \"Abundance\", \"id\": \"b9dc79a894f3563277e1780335b6b791671aed2b5c99c4cff346dd7dc034cf7a2f45aff6aa7a6c5a68b004a8a90a7cc651a36c5faedbac5019bc83f21261302d\", \"name\": \"Tau antibody, 40E8\", \"namespace\": \"HBP\"}, {\"function\": \"Abundance\", \"id\": \"bce171e6abc27abe31bbf906c70cafa1a52e04542d8f15aca67aa1a53cd2ad9a3fbde4bee5c0d4128c35953c56c45a11f9e57de3c2359edf498b1acfae246eab\", \"name\": \"Tau antibody, pS396\", \"namespace\": \"HBP\"}, {\"function\": \"Abundance\", \"id\": \"38e1843a906b6279ca2a95c5dc6dc834704e57c748aa0b4f88f0c8100d80f1a6e9fdb1cb9085ba0caebac6ca3fd53d022433d91ed309b03dd9287f7f56f57aae\", \"name\": \"(S)-2-ethyl-8-methyl-1-thia-4,8-diazaspiro(4.5)decan-3-one\", \"namespace\": \"MESH\"}, {\"function\": \"Abundance\", \"id\": \"ab9ab8a42f27ea8b317e817b34a7ed7609808e3ccad71f0ce8685fa7b9e53fd847534ce7483452cc11f2166657f178e8e6804466c34019ef9674372b725b155d\", \"name\": \"Receptors, Nicotinic\", \"namespace\": \"MESH\"}, {\"function\": \"BiologicalProcess\", \"id\": \"f1a3202be460e565f65c1445ee061e99f63cc05f9ab70c5d4d8d93ba180d6a94f65cb8d1ac6c5ba53c0a4d2e153db32135e6a0bffcfdfd4d84e6727594a70b0a\", \"name\": \"autophagy\", \"namespace\": \"GO\"}, {\"function\": \"BiologicalProcess\", \"id\": \"568e5f70e05a1543c9a1d54019c461ab0802ff3556801ad06f2231e0687a80f2cbd21d12f808f5d72752f7cc8c484e0b32d830256321af323cfd08b1b63cc765\", \"name\": \"macroautophagy\", \"namespace\": \"GO\"}, {\"function\": \"Protein\", \"id\": \"a85beb5b2b9417fabbbcc2135b9533cb0766b38688caf02242632843ae455bad332fb625d6e8296ff77b40e841485a56854fc14e300552425d596e8a8e70fd3f\", \"name\": \"CDK5\", \"namespace\": \"HGNC\"}, {\"function\": \"Protein\", \"id\": \"962b8a84d274a6e78416261efc2137bcf03c8035a7be3138736624f9c08fe9ae92738ac20d9edd54a6ace6ff161f28793df367f2e084e2837fd799e530fddbbd\", \"name\": \"CHRM1\", \"namespace\": \"HGNC\"}, {\"function\": \"Protein\", \"id\": \"ac819ecac7786e3ea9b6906b318167c6a94fbbf428aff46f85083b83c50b9604342d39c1c36ff49c2cb3a04e491913164c3e4e7889dedd513b6196d9b0560c7c\", \"name\": \"GSK3B\", \"namespace\": \"HGNC\"}, {\"function\": \"Protein\", \"id\": \"f618f39c33e74791dcba1fbe9d727a07c9eb2f3e0b415a6d7752e89ed99787c53592b44bf93bb6f6d1f5ea66a454795a4ad9b749ea7c0c19d2b2bcc10d646b0e\", \"name\": \"MAPT\", \"namespace\": \"HGNC\", \"variants\": [{\"identifier\": {\"name\": \"HBP00007\", \"namespace\": \"HBP\"}, \"kind\": \"pmod\"}]}, {\"function\": \"Protein\", \"id\": \"ede67fb639f29861d91976fefa16d8279b53ae13a659f35d70ed12e01bf3f339ffcf701fc54c7c4e908fee9902c02aac3e0022dcae5a7afa217ed2d4dba133b0\", \"name\": \"MAPT\", \"namespace\": \"HGNC\", \"variants\": [{\"identifier\": {\"name\": \"hyperphosphorylation\", \"namespace\": \"HBP\"}, \"kind\": \"pmod\"}]}, {\"function\": \"Protein\", \"id\": \"526bae251987f8c4f55f6e2451662fa55e087c57585beae7c389b020d73487c739834e9b8ec841783de6a5b5fb9df8b138e9eff4406a9ec5e606766575dcac0e\", \"name\": \"MAPT\", \"namespace\": \"HGNC\", \"variants\": [{\"identifier\": {\"name\": \"Ph\", \"namespace\": \"bel\"}, \"kind\": \"pmod\"}]}, {\"function\": \"Protein\", \"id\": \"df7a2c04f4438d51f2e9e042ebd8ae9456487148c38b1ee0deee645865eac74380f664a645c8c16f21cf9900e7a379af8b9e9c4a72dd57b7d1e0ba9738364fb0\", \"name\": \"NPAS4\", \"namespace\": \"HGNC\"}, {\"function\": \"Protein\", \"id\": \"d2586deea41500a6ea7755a94475bef3e256337b994748bb58034e8a43261cfea2cf38bc85241d08d9e06bdfc15b92df32297bceb6a3bf8848c3f2d6878b030b\", \"name\": \"TFEB\", \"namespace\": \"HGNC\"}, {\"function\": \"Protein\", \"id\": \"ee9a8850def1c6c0a93482d8d5c04086543f33fe5ab23c7c0f667149ea6d1b7b2eec7dfa67785a7199d05fff966640fc48cdd54d49b47adf5af492c9addf8337\", \"name\": \"Cyclic AMP-Dependent Protein Kinases\", \"namespace\": \"MESH\"}, {\"function\": \"Pathology\", \"id\": \"3196cc4c6a20279c822cae925a9b271d0224f2c8d468e3069a3ee460662be286b61db2f144c550686a9c624e58af4ffa7570f4f2384e1cb8ee40496f8cb3fa5e\", \"name\": \"Inflammation\", \"namespace\": \"MESH\"}, {\"function\": \"Pathology\", \"id\": \"0893d59040331bc1b155187396067458820ab2635e0138b840323082d33d44b964ab1240f5aeb64c5360e1340de40545835bc744c181abb910d64c923d55ff2b\", \"name\": \"Alzheimer Disease\", \"namespace\": \"MESHD\"}]}, \"#zvtuopsqryxfgcbe\", 1000, 650, {'Protein': '#1F77B4', 'Pathology': '#FF7F0E', 'BiologicalProcess': '#2CA02C', 'miRNA': '#D62728', 'Complex': '#98DF8A', 'Composite': '#9467BD', 'Reaction': '#000000', 'Gene': '#FFBB78', 'Abundance': '#AEC7E8', 'RNA': '#FF9896'});\n",
       "});"
      ],
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mapt_ptm_controller_graph = get_upstream_causal_subgraph(graph, nodes)\n",
    "mapt_ptm_controller_graph.name = 'MAPT Post-Translational Modification Controllers'\n",
    "mapt_ptm_controller_graph.version = hbp_knowledge.VERSION\n",
    "mapt_ptm_controller_graph.summarize()\n",
    "to_jupyter(x)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get a list of controller proteins."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['CHRM1', 'CDK5', 'GSK3B', 'NPAS4', 'TFEB']\n"
     ]
    }
   ],
   "source": [
    "controllers = [\n",
    "    node.name\n",
    "    for node in mapt_ptm_controller_graph\n",
    "    if isinstance(node, Protein) and 'MAPT' != node.name and 'HGNC'== node.namespace.upper()\n",
    "]\n",
    "\n",
    "print(controllers)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TODO\n",
    "\n",
    "\n",
    "## INDRA Enrichment\n",
    "\n",
    "1. Generate query to find statements that have MAPT as an object and use a conversion statement\n",
    "2. Get bigger list of \n",
    "\n",
    "## Visualization of Relevant Chemical Space\n",
    "\n",
    "1. Load ExCAPE-DB\n",
    "2. Extract relationships from ExCAPE-DB related to these proteins\n",
    "3. Calculate chemical descriptors\n",
    "4. Show a t-SNE plot or related 2D depiction of chemical space"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
